# LRT_microbiomes_critical_illness

This repository contains data and analytical code for bioinformatic analyses of the manuscript entitiled "Deep longitudinal lower respiratory tract microbiome profiling reveals genome-resolved functional and evolutionary dynamics in critical illness" (under revision, Nature Communications).

## Software required
- `ggplot2`: 3.4.1
- `ggtext`: 0.1.2
- `ggstar`: 1.0.4 
- `ComplexHeatmap`: 2.12.1 
- `stringr`: 1.5.0
- `aplot`: 0.1.10
- `ggeasy`: 0.1.4 
- `Cairo`: 1.6.0
- `circlize`: 0.4.15
- `scales`: 1.2.1
- `plotly`: 4.10.1
- `tidyverse`: 2.0.0 
- `reshape2`: 1.4.4
- `readxl`: 1.4.2
- `genoPlotR`: 0.8.11
- `vegan`: 2.6.4
- `ggtreeExtra`: 1.6.0 
- `ggtree`: 3.7.1.3
- `treeio`: 1.20.1
- `tidytree`: 0.4.2
- `viridis`: 0.6.2
- `ggnewscale`: 0.4.8
- `PCAtools`: 2.8.0
- `phyloseq`: 1.40.0
- `emmeans`: 1.8.5 
- `ggsci`: 3.0.0
- `tidyHeatmap`: 1.8.1
- `ape`: 5.7.1
- `magrittr`: 2.0.3 
- `pheatmap`: 1.0.12

The analytical code provided in this repository was implemented using R version 4.2.0 in RStudio. No specialized hardware requirements are necessary to run the code.

## Data and Code
- The directory 'Data_repository' contains the data required for the analyses and visualization of results.
- The directory 'Scripts' contains the code required for the analyses and visualization of results.
