## Libraries

library(ggVennDiagram)
library(VennDiagram)
library(ggplot2)
library(ggpubr)
library(ggvenn)
library(dplyr)
library(patchwork)

## Input data
# input data was stored at the directory 'Data_repository'



```{r Fig 1b}

df_metaphlan_all <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Main_figures\microbiome_species-top_prevalent.csv)')
df_metaphlan_all[df_metaphlan_all$SamplingSite=='HospitalA',]$Name
df_input_list <- list(
  "Hospital A" = df_metaphlan_all[df_metaphlan_all$SamplingSite=='Hospital-A',]$taxonomy,
  "Hospital B" = df_metaphlan_all[df_metaphlan_all$SamplingSite=='Hospital-B',]$taxonomy,
  "Hospital C" = df_metaphlan_all[df_metaphlan_all$SamplingSite=='Hospital-C',]$taxonomy
)

venn.diagram(
  x = df_input_list,
  category.names = c("Hospital A" , "Hospital B" , "Hospital C"),
  filename = r'()', # fill the filepath where you want to save the plot
  output=TRUE,
  
  # Output features
  imagetype="png" ,
  height = 1500 , 
  width = 1500 , 
  resolution = 900,
  compression = "lzw",
  
  # Circles
  lwd = 0.7,
  # col=c("#440154ff", '#509fd2', '#5da94e'),
  col=c("#d93526", "#f7a902", "#1a6196"),
  # fill = c(alpha("#440154ff",0.4), alpha('#509fd2',0.4), alpha('#5da94e',0.4)),
  fill = c(alpha("#d93526",0.3), alpha('#f7a902',0.3), alpha('#1a6196',0.3)),
  cex = 0.7,
  fontfamily = "sans",
  cat.cex = 0.6,
  cat.default.pos = "outer",
  cat.pos = c(-27, 27, 135),
  cat.dist = c(0.055, 0.055, 0.085),
  cat.fontfamily = "sans",
  # cat.col = c("#440154ff", '#509fd2', '#5da94e'),
  cat.col = c("#d93526", "#f7a902", "#1a6196"),
  rotation = 1
)

```

```{r Fig 1c}

df_prevalent <- read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Main_figures\microbiome_species-top_prevalent.csv)")
df_order <- read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Main_figures\microbiome_species-top_prevalent-sorted.csv)")
df_sorted_desc <- arrange(df_order, desc(prevalency_percentage_all))

order_species <- df_sorted_desc[1:10,'taxonomy']
df_prevalent$taxonomy <- factor(df_prevalent$taxonomy, levels = df_sorted_desc[1:10,'taxonomy'])

#transform species name into human-readable format
df_prevalent$taxonomy <- gsub("^s__", "", df_prevalent$taxonomy) # Remove the prefix
df_prevalent$taxonomy <- gsub("_", " ", df_prevalent$taxonomy)
plot_df <- ggplot(na.omit(df_prevalent), aes(fill=SamplingSite, x=reorder(taxonomy,prevalency_percentage,sum,decreasing=T), y=prevalency_percentage))+  #'x=reorder(species,prevalency_percentage,sum,decreasing=T)' --> order stacked barplot with sum of values
  geom_bar(position = "dodge",stat = "identity",) +
  scale_fill_manual(values = c('Hospital-A'='#d93526','Hospital-B'='#f7a902','Hospital-C'='#1a6196'))+
  theme_pubr(x.text.angle = 35)

plot_df
ggpar(plot_df,
      ylab = "Percentage of samples with\ndetected species (%)",
      xlab = "Species",
      font.xtickslab=c(18),
      font.ytickslab=c(18),
      font.x = c(18,"bold"),
      font.y = c(18,"bold"),
      font.legend = c(18,'bold'),
      x.text.angle =50) + theme_pubr(base_size = 24,x.text.angle =35,legend = "none",margin = FALSE)

```

```{r Fig 1d}

```

```{r Fig 1e}

```

```{r Fig S1a}

#most prevalent species for hospitalA
df_species_hospitalA <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\microbiome_species-top_prevalent.csv\microbiome_species-statistics_count-Hospital-A.csv)',header = TRUE) # save in the same file 'microbiome_species-statistics_count-Hospitals.txt'
df_species_hospitalA$taxonomy <- gsub("^s__", "", df_species_hospitalA$taxonomy) # Remove the prefix
df_species_hospitalA$taxonomy <- gsub("_", " ", df_species_hospitalA$taxonomy)

plot_species_hospitalA <- ggbarplot(df_species_hospitalA[1:5,],x = "taxonomy", y = "prevalency_percentage",orientation = "vertical",alpha=1,
                                    fill = "#d93526", color = "#d93526",sort.val = 'desc',ggtheme = theme_bw(base_size = 18)) + 
  labs(x=NULL,y='Percentage of\ndetected samples (%)') +  theme(aspect.ratio =1/1.5 )  #+ #+ ylim(0,15) 
  #scale_y_continuous(breaks = c(0,2,4,6,8,10))

plot_species_hospitalA <-ggpar(plot_species_hospitalA,x.text.angle = 55)

#-------------------------------------------------------------------------------------------------------------------------------

#most prevalent species for hospitalB
df_species_hospitalB <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\microbiome_species-top_prevalent.csv\microbiome_species-statistics_count-Hospital-B.csv)',header = TRUE # save in the same file 'microbiome_species-statistics_count-Hospitals.txt'
df_species_hospitalB$taxonomy <- gsub("^s__", "", df_species_hospitalB$taxonomy) # Remove the prefix
df_species_hospitalB$taxonomy <- gsub("_", " ", df_species_hospitalB$taxonomy)

plot_species_hospitalB <- ggbarplot(df_species_hospitalB[1:5,],x = "taxonomy", y = "prevalency_percentage",orientation = "vertical",alpha=1,
                                    fill = "#f7a902", color = "#f7a902",sort.val = 'desc',ggtheme = theme_bw(base_size = 18)) + 
  labs(x='Species')  +  theme(aspect.ratio =1/1.5,axis.title.y = element_blank())

plot_species_hospitalB <- ggpar(plot_species_hospitalB,x.text.angle = 55)

#-------------------------------------------------------------------------------------------------------------------------------

#most prevalent species for hospitalC
df_species_hospitalC <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\microbiome_species-top_prevalent.csv\microbiome_species-statistics_count-Hospital-C.csv)',header = TRUE) # save in the same file 'microbiome_species-statistics_count-Hospitals.txt'
df_species_hospitalC$taxonomy <- gsub("^s__", "", df_species_hospitalC$taxonomy) # Remove the prefix
df_species_hospitalC$taxonomy <- gsub("_", " ", df_species_hospitalC$taxonomy)
df_species_hospitalC <- df_species_hospitalC[-5,]

plot_species_hospitalC <- ggbarplot(df_species_hospitalC[1:5,],x = "taxonomy", y = "prevalency_percentage",orientation = "vertical",alpha=1,
                                    fill = "#1a6196", color = "#1a6196",sort.val = 'desc',ggtheme = theme_bw(base_size = 18)) + 
  labs(x=NULL)  +  theme(aspect.ratio =1/1.5,axis.title.y = element_blank() )

plot_species_hospitalC <-ggpar(plot_species_hospitalC,x.text.angle = 55)

plot_species_hospitalA|plot_species_hospitalB|plot_species_hospitalC

```

```{r Fig S1b}

df_too_low <- read.csv((r'(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\microbiome_species-top_prevalent.csv\head-to-head_comparisons-DNA_concentration_too_low.csv)'))
df_too_low$Percentage_Failed <- df_too_low$Percentage_Failed *100
df_too_low$Percentage_successful <- 100 - df_too_low$Percentage_Failed
plot <- ggbarplot(df_too_low, "Method", "Percentage_successful",
                  color = '#1f78b4',fill='#1f78b4',palette = "#1f78b4",
                  label = F,
                  position = position_dodge(0.9)) + 
  scale_x_discrete(labels= c('CMEM','Power Water','AllPrep DNA')) + 
  labs(x=NULL,y='Percentage of samples with detectable\nDNA after extraction (%)') + theme_bw(base_size = 27)
ggpar(plot,legend = 'bottom',x.text.angle = 30)

```

```{r Fig S1c}

df_concentration <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\head-to-head_comparisons-DNA_concentration.csv)')
df_concentration$Concentration  <- log10(df_concentration$Concentration + 1)
df_concentration$Method <- factor(df_concentration$Method,levels =c('CMEM','Power Water','AllPrep DNA') )
my_comparisons <- list(c('CMEM','Power Water'),c('CMEM','AllPrep DNA'),c('Power Water','AllPrep DNA'))
p_concentration <- ggboxplot(df_concentration, "Method", "Concentration",
          color = "Method", palette =c("#00AFBB", "#E7B800", "#FC4E07"),
          add = "jitter",add.params = list(size = 1.5, alpha = 0.5))  +
  stat_boxplot(geom = "errorbar",width=0.05,color=c("#00AFBB", "#E7B800", "#FC4E07"))  +
  stat_compare_means(comparisons = my_comparisons,label = "p.label",size=8,vjust=0) +
  scale_x_discrete(labels= c('CMEM','Power Water','AllPrep DNA')) + 
  labs(x=NULL,y='DNA concentration (log10, ng/ul)',) + #ylim(-1,10) +
  theme_bw(base_size = 24)

```

```{r Fig S1d}

```

```{r Fig S1e}
df_profiles_vegan  <-  read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\microbial_profile-feasibility_validation.csv)",header = TRUE)
metadata = read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\metadata-microbial_profile-feasibility_validation.csv)",header = TRUE) 

df_profiles_vegan_abundance <- df_profiles_vegan[,3:ncol(df_profiles_vegan)]
df_profiles_vegan_matrix <- as.matrix(df_profiles_vegan[,3:ncol(df_profiles_vegan)])

abundance <- select(df_profiles_vegan,-c(SampleID,Group))
rownames(abundance) <- df_profiles_vegan[1:nrow(df_profiles_vegan),1]
abundance <- t(abundance) #transposed dataframe

pca_metadata <- data.frame(df_profiles_vegan[1:nrow(df_profiles_vegan),2])
rownames(pca_metadata) <- df_profiles_vegan[1:nrow(df_profiles_vegan),1]
colnames(pca_metadata) <- c('Group')
p <- pca(abundance,metadata=pca_metadata)

# adonis test 
set.seed(123)
adonis2(df_profiles_vegan_matrix ~ Group, df_profiles_vegan, method = "bray", permutations = 999)

df_pca_output <- as.data.frame(p$rotated)
df_pca_output$SampleID <- rownames(df_pca_output)
df_pca_output <- merge(df_pca_output, metadata, by = "SampleID")

ggscatter(df_pca_output, x = "PC1", y = "PC2", color = "Group", palette = c("#0466AB", "#D15D4E"), 
           size =5, alpha = 1) +
  geom_line(data = df_pca_output, aes(group = Order), color = "black",alpha = 0.8) +
  # stat_ellipse(aes(color = Group), geom = "path", alpha = 1, linetype = "solid",linewidth =0.8,
  #              level = 0.9) + theme_pubr(border = T)
  stat_ellipse(aes(fill = Group), geom = "polygon", linetype = "solid", level = 0.95,alpha=0.3) +
#   scale_fill_manual(values = c("#0466AB", "#FFEFF2"))
  theme_pubr(base_size = 23,border = T)

```

```{r Fig S1f}

df_diversity  <- read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\alpha_diversity-comparison.csv)",header = TRUE)
df_diversity$Group  <- factor(df_diversity$Group,levels = c("Undepleted","Depleted"))


my_comparisons <- list(c("Undepleted","Depleted"))

p_diversity_shannon <- ggplot(df_diversity, aes(x=Group,y=Alpha_diversity_Shannon,fill=Group)) +
  stat_boxplot(geom = "errorbar",width=0.15)+
  geom_boxplot(size=0.5 ) + #theme_classic2() +
  geom_line(aes(group=Order),colour='Gray75', size=0.7)+
  geom_line(size=0.3) +
  scale_fill_manual(values = c("#00AFBB", "#E7B800")) +
  scale_x_discrete(labels = c("Control","Treatment")) +
  stat_compare_means(paired = T,label.y = 2.2,comparisons = my_comparisons,label = "p.signif",size=7,bracket.size = 0.5) +
  stat_compare_means(paired = T,label.y = 2.5,label.x = 1,size=8) + 
  labs(x = NULL, y = "Alpha diversity (Shannon)") +
  theme_pubr(base_size = 20, border =F,legend = "none")

p_diversity_simpson <- ggplot(df_diversity, aes(x=Group,y=Alpha_diversity_Simpson,fill=Group)) +
  stat_boxplot(geom = "errorbar",width=0.15)+
  geom_boxplot(size=0.5 ) + #theme_classic2() +
  geom_line(aes(group=Order),colour='Gray75', size=0.7)+
  geom_line(size=0.3) +
  scale_fill_manual(values = c("#00AFBB", "#E7B800")) +
  scale_x_discrete(labels = c("Control","Treatment")) +
  stat_compare_means(paired = T,label.y = 0.9,comparisons = my_comparisons,label = "p.signif",size=7,bracket.size = 0.5) +
  stat_compare_means(paired = T,label.y = 1.1,label.x = 1,size=8) + 
  labs(x = NULL, y = "Alpha diversity (Simpson)") +
  theme_pubr(base_size = 20, border =F,legend = "none")

```

```{r Fig S1g}

df_total_args  <- read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\total_arg_abundance-comparison.csv)",header = TRUE)
df_total_args$Group  <- factor(df_total_args$Group,levels = c("Undepleted","Depleted"))
df_total_args$total_arg_abundance_log10 <- log10(df_total_args$total_arg_abundance)
my_comparisons <- list(c("Undepleted","Depleted"))
p_arg_abundance <- ggplot(df_total_args, aes(x=Group,y=total_arg_abundance_log10,fill=Group)) +
  stat_boxplot(geom = "errorbar",width=0.15)+
  geom_boxplot(size=0.5 ) + #theme_classic2() +
  geom_line(aes(group=Order),colour='Gray75', size=0.7)+
  geom_line(size=0.3) +
  scale_fill_manual(values = c("#00AFBB", "#E7B800")) +
  scale_x_discrete(labels = c("Control","Treatment")) +
  stat_compare_means(paired = T,label.y = 1.8,comparisons = my_comparisons,label = "p.signif",size=7,bracket.size = 0.5) +
  stat_compare_means(paired = T,label.y = 2.5,label.x = 1,size=8) + 
  labs(x = NULL, y = "Total ARG abundance (log10)") +
  theme_pubr(base_size = 20, border =F,legend = "none")

```

```{r Fig S1h}

df_diversity  <- read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\alpha_diversity-metaphlan_speices-all_patients.csv)",header = TRUE)
df_metadata_all  <- read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\metaData.csv)",header = TRUE)

### merge the metadata and alpha diversity data

df_subset <- select(df_metadata_all, SampleID, Diagnosed.Pneumonia, Antibiotic.Use)

df_merged <- merge(df_diversity, df_subset, by = "SampleID")
df_merged_subset <- df_merged[df_merged$Diagnosed.Pneumonia %in% c("Yes", "No"), ]
df_merged_subset$Diagnosed.Pneumonia   <- factor(df_merged_subset$Diagnosed.Pneumonia,levels = c("No","Yes"))
my_comparisons <- list(c("No",'Yes'))

p_diversity_shannon <- ggplot(df_merged_subset, aes(x=Diagnosed.Pneumonia,y=Alpha_diversity_Shannon,fill=Diagnosed.Pneumonia)) +
  stat_boxplot(geom = "errorbar",width=0.15)+
  geom_boxplot(size=0.5 ) + #theme_classic2() +
  geom_line(size=0.3) +
  scale_fill_manual(values = c("#00AFBB", "#E7B800")) +
  scale_x_discrete(labels = c("No Pneumonia","Pneumonia")) +
  stat_compare_means(label.y = 3.2,comparisons = my_comparisons,label = "p.signif",size=7,bracket.size = 0.5) +
  stat_compare_means(label.y = 3.8,label.x = 1,size=8) + 
  labs(x = NULL, y = "Alpha diversity (Shannon)") +
  theme_pubr(base_size = 21, border =F,legend = "none")
ggpar(p_diversity_shannon,x.text.angle = 30,legend = 'none')

```

```{r Fig S1i}

df_arg_abundance  <- read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\argtypes-pneumonia_vs_nopneumonia.csv)",header = TRUE)
df_arg_abundance$totoal_arg_abundance_log10 <- log10(df_arg_abundance$totoal_arg_abundance)
my_comparisons <- list(c("No",'Yes'))
p_arg_abundance <- ggplot(df_arg_abundance, aes(x=Diagnosed.Pneumonia,y=totoal_arg_abundance_log10,fill=Diagnosed.Pneumonia)) +
  stat_boxplot(geom = "errorbar",width=0.15)+
  geom_boxplot(size=0.5 ) + #theme_classic2() +
  geom_line(size=0.3) +
  scale_fill_manual(values = c("#00AFBB", "#E7B800")) +
  scale_x_discrete(labels = c("No Pneumonia","Pneumonia")) +
  stat_compare_means(label.y = 3.0,comparisons = my_comparisons,label = "p.signif",size=7,bracket.size = 0.5) +
  stat_compare_means(label.y = 3.8,label.x = 1,size=8) + 
  labs(x = NULL, y = "Total ARG abundance (log10)") +
  theme_pubr(base_size = 20, border =F,legend = "none")
ggpar(p_arg_abundance,x.text.angle = 30,legend = 'none')

```

```{r Fig 2a}

df_profiles = read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Main_figures\microbiota-pneumonia-species_phyloseqized.csv)",header = FALSE)
sampleid2site = read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Main_figures\metadata-pneumonia-sampleid2site.csv)",header = TRUE)
df_arrows <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Main_figures\MDS_metrics-bray-specie_pneumonia-arrows_coordinates.csv)')

#otu-like data reformat 
otu_metaphlan <- as.matrix(df_profiles[2:nrow(df_profiles),2:ncol(df_profiles)]) # otu matrix can't have any character-like data, you have to use index to remove it
class(otu_metaphlan) <- "numeric" # convert the datatype of the matrix into "numeric"
otu_metaphlan <- otu_metaphlan*1000000
rownames(otu_metaphlan) <- df_profiles[2:nrow(df_profiles),1]
colnames(otu_metaphlan) <- df_profiles[1,2:ncol(df_profiles)]

OTU = otu_table(otu_metaphlan,taxa_are_rows = TRUE)
rownames(sampleid2site) <- sampleid2site$SampleID # you have to re-assign the rownames of meta-data, or you will run into a problem: "invalid “phyloseq” object: Component sample names do not match"
sam_data = sample_data(sampleid2site)
physeq_profiles = phyloseq(OTU,sam_data) 
target_dist_methods <- c('bray')
for( i in target_dist_methods ){
  target_dist <- distance(physeq_profiles, method=i)
  target_MDS_metrics <- ordinate(physeq_profiles, "MDS", distance=target_dist)
  df_metrics <- data.frame(target_MDS_metrics[["vectors"]])
  df_metrics_filter <- df_metrics[,1:2]
  
  # Merge df_metrics_filter and sampleid2site on the index
  df_metrics_filter <- merge(df_metrics_filter, sampleid2site, by="row.names", all.x=TRUE)
  df_metrics_filter  <- df_metrics_filter[,c(4,2,3,5)]
}

df_arrows$variable <- str_split(df_arrows$variable, "__", simplify = TRUE)[,2]
df_arrows$variable  <- str_replace_all(df_arrows$variable, "_", " ")
df_arrows$variable  <- str_replace_all(df_arrows$variable, " group", " ")
df_distance_bray$Axis.2 <- df_distance_bray$Axis.2*-1
df_arrows$Axis.2 <- df_arrows$Axis.2*-1

plots <- ggscatter(df_distance_bray,
                   x = "Axis.1", 
                   y = "Axis.2",
                   color = "SamplingSite",
                   #shape = "SamplingSite",
                   size=6,
                   # palette = c( "#FC4E07","#00AFBB", "#E7B800"), #color1
                   palette = c("#e64833", "#01a188", "#3e5386"), #color2
                   # palette = c("#c23726", "#e3e3e3", "#0036fc"),  #palette2
                   # palette = c( "#010101","#fe0000", "#0037fe"), #palette4
                   #palette = c("#c23726", "#5a5a5a", "#2c83c6"),
                   #palette = c("#7152a9", "#ffc376", "#eb7600"),
                   margin.plot = "boxplot",
                   ggtheme = theme_pubr(),
                   #margin.params = list(fill="SamplingSite",size=0.36),
                   legend = 'right') + 
  geom_segment(data = df_arrows,
               x = 0, y = 0, alpha = 0.7,
               mapping = aes(xend = Axis.1, yend = Axis.2),
               color = 'black',size=0.1,
               # Add arrow head
               arrow = arrow(length = unit(3, "mm"))) +
  geom_label(data = df_arrows, aes(label = variable),color='black',alpha=0.7,size=7,label.size =0.01) 
ggpar(plots,
      xlab = "MDS1 (27.6%)",
      ylab = "MDS2 (16.6%)") + theme_pubr(base_size = 24,border = T,legend = 'none')

```

```{r Fig 2b}

df_distance_bray_species_pneumonia <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Main_figures\MDS_metrics-bray+species_abundance-species_pneumonia.csv)')
df_distance_bray_species_pneumonia$variable <- str_split(df_distance_bray_species_pneumonia$variable, "__", simplify = TRUE)[,2]
df_distance_bray_species_pneumonia$variable  <- str_replace_all(df_distance_bray_species_pneumonia$variable, "_", " ")
df_distance_bray_species_pneumonia$variable  <- str_replace_all(df_distance_bray_species_pneumonia$variable, " group", " ")
df_distance_bray_species_pneumonia$Axis.2 <- df_distance_bray_species_pneumonia$Axis.2*-1

plots_mds_species <- ggscatter(df_distance_bray_species_pneumonia,
                   x = "Axis.1", 
                   y = "Axis.2",
                   # color = "Streptococcus",
                   color = 'value',
                   #shape = "SamplingSite",
                   size=6,
                   margin.plot = "boxplot",
                   ggtheme = theme_pubr(border = T),
                   #margin.params = list(fill="Acinetobacter",size=0.36),
                   legend = 'right') + #gradient_color(inferno(10))
                   gradient_color(c('#bfbdbd','#fe1608')) + facet_wrap(~ variable) +
  labs(x='MDS1 (19.8%)',y='MDS2 (14.5%)',color='Relative\nabundance (%)') + 
  theme(axis.text.x  = element_text(size=32),axis.text.y = element_text(size=32),
        axis.title.x = element_text(size = 32),axis.title.y = element_text(size = 32)) +
  theme(strip.background = element_blank(),
        strip.text = element_textbox(
          size = 30,
          color = "black", fill = "#d9d9d9", box.color = "#4A618C",#face = 'italic',
          halign = 0.5, linetype = 1,  width = unit(1, "npc"),
          margin = margin(0, 0, 0, 0),padding = margin(6, 6, 6, 6)),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.x = element_blank(),
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank(),
        legend.position = 'none',
        legend.title = element_text(size=14),
        legend.text = element_text(size=16)) +
  xlim(-0.5,0.53)
plots_mds_species

```

```{r Fig 2c,d}

patientids <- c('P68','P23')
df_abundance <- read.csv(paste("LRT_microbiomes_critical_illness\Data_repository\Main_figures/microbiota-individual_patient/",
                               patientid,"-metaphlan3-species_microbiota.csv",sep=""))

df_abundance$Sampling_Day_ICU <- factor(df_abundance$Sampling_Day_ICU,levels = unique(df_abundance$Sampling_Day_ICU))
plot_df <- ggplot(df_abundance, aes(fill=taxonomy, x=Sampling_Day_ICU, y=relative_abundance))+
  geom_bar(position = "stack",stat = "identity",) + ggtitle(label = patientid ) + 
  scale_fill_manual(values = species2color)
ggpar(plot_df,
      ylab = "Relative\nabundance (%)",
      xlab = F) + theme_pubr(base_size = 14,x.text.angle =50,legend = "none",margin = FALSE)  + xlab(NULL) +
  ggeasy::easy_center_title() 

```

```{r Fig 2e}

### visualized with circos webserver

```

```{r Fig 2g}

my_comparisons <- list(c("Yes", "No"))
df_combined$tetracycline__tetA_log10 <- log10(df_combined$tetracycline__tetA)
ggboxplot(df_combined, x="Diagnosed.Pneumonia", y="tetracycline__tetA_log10",
          color = "Diagnosed.Pneumonia", 
          add = "jitter",add.params = list(size = 2.4, alpha = 0.8))  +
  geom_boxplot(aes(color=Diagnosed.Pneumonia), lwd = 0.7) +
  scale_x_discrete(labels = c("No" = "No Pneumonia", "Yes" = "Pneumonia")) +
  stat_compare_means(comparisons = my_comparisons,label = "p.label",size=7,
                     vjust=.15,tip.length= c(0,0,0)) + 
  labs(x=NULL,y='Total ARG abundance (log10)',) +  ggtitle('Tetracycline__tetA') +
  # scale_y_continuous(limits=c(-2,2))+
  theme_pubr(base_size = 23,legend ='none',x.text.angle = 30) + 
  ggeasy::easy_center_title() + 
  scale_color_npg()

```

```{r Fig 2h}

my_comparisons <- list(c("Yes", "No"))

df_combined$multidrug__mexB_log10 <- log10(df_combined$multidrug__mexB)
ggboxplot(df_combined, x="Diagnosed.Pneumonia", y="multidrug__mexB_log10",
          color = "Diagnosed.Pneumonia", 
          add = "jitter",add.params = list(size = 2.4, alpha = 0.8))  +
  geom_boxplot(aes(color=Diagnosed.Pneumonia), lwd = 0.7) +
  scale_x_discrete(labels = c("No" = "No Pneumonia", "Yes" = "Pneumonia")) +
  stat_compare_means(comparisons = my_comparisons,label = "p.label",size=7,
                     vjust=.15,tip.length= c(0,0,0)) + 
  labs(x=NULL,y='Total ARG abundance (log10)',) +  ggtitle('Multidrug__mexB') +
  theme_pubr(base_size = 23,legend ='none',x.text.angle = 30) + 
  ggeasy::easy_center_title() + 
  scale_color_npg()

```

```{r Fig 2i}

```

```{r Fig 2j}

```

```{r Fig S2a}

df_species_selected <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\MDS_metrics-bray+species_abundance-species_pneumonia.csv)')
df_species_selected$SamplingSite  <- str_replace_all(df_species_selected$SamplingSite, "-", "")
df_species_selected$variable <- str_split(df_species_selected$variable, "__", simplify = TRUE)[,2]
df_species_selected$variable  <- str_replace_all(df_species_selected$variable, "_", " ")
df_species_selected$variable  <- str_replace_all(df_species_selected$variable, " group", " ")

my_comparisons <- list( c("HospitalA", "HospitalB"), c("HospitalA", "HospitalC"), c("HospitalB", "HospitalC") )
df_species_selected$SamplingSite <- factor(df_species_selected$SamplingSite,levels = c("HospitalA", "HospitalB","HospitalC"))

p_species_selected <- ggboxplot(df_species_selected, "SamplingSite", "value",
                             color = "SamplingSite", palette =c("#e64833", "#01a188", "#3e5386"),
                             add = "jitter",add.params = list(size = 2, alpha = 0.5))  +
  stat_boxplot(mapping = aes(color=SamplingSite),geom = "errorbar",width=0.05)  +
  stat_compare_means(comparisons = my_comparisons,label = "p.label",
                     size=5.0,vjust=.1,tip.length= c(0,0,0)) + ylim(-1,135) +
  facet_grid(~variable) + labs(x=NULL,y='Relative abundance (%)') +
  theme_pubr(base_size = 15,border = T,legend ='none',x.text.angle = 30) +
  theme(strip.text.x = element_text(size = 14))      
p_species_selected

```

```{r Fig S2b}

df_top_pathabundance <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\humann3-top_pathabundance-each_hospital-merged_top.csv)')
df_metadata <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\humann3_metadata.csv)')
matrix_top_abundance <- as.matrix(df_top_pathabundance[,-1])
row.names(matrix_top_abundance) <- df_top_pathabundance$pathway
matrix_top_abundance_log <- log10(matrix_top_abundance)
matrix_top_abundance_log[is.infinite(matrix_top_abundance_log)] <- 0
plot_top_pathabundance <- Heatmap(matrix_top_abundance_log,show_column_names = T,show_row_names = T,cluster_columns = F,cluster_rows = T,
        rect_gp = gpar(col= "grey"),name='Abundance (log10)',
        row_names_gp = gpar(fontsize=20),column_names_gp = gpar(fontsize=20),
        heatmap_legend_param = list(title_gp = gpar(fontsize=20),labels_gp = gpar(fontsize=20)))

```

```{r Fig S2c}

df_tempral_dyanmics_group_new_criteria <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\OI+WBC+Neutrophils-temporal_dynamics_groups.csv)')
df_tempral_dyanmics_group_new_criteria$Group <- factor(df_tempral_dyanmics_group_new_criteria$Group,levels =c('multiple','stable') )
df_tempral_dyanmics_groups_no_na <- na.omit(df_tempral_dyanmics_groups)
my_comparisons <- list(c('multiple','stable'))
df_tempral_dyanmics_group_new_criteria_OI <- df_tempral_dyanmics_group_new_criteria[!is.na(df_tempral_dyanmics_group_new_criteria$Oxygenation.Index.of.The.Day), ]

ggboxplot(df_tempral_dyanmics_group_new_criteria_OI, x="Group", y="Oxygenation.Index.of.The.Day",
          color = "Group",palette =c("#E8080A", "#0A74B2"),
          add = "jitter",add.params = list(size =2.4, alpha = 0.8))  +
  geom_boxplot(aes(color=Group), lwd = 0.7) +
  scale_x_discrete(labels = c("multiple" = "Multi-dominant", "stable" = "Mono-dominant")) +
  stat_compare_means(comparisons = my_comparisons,label = "p.label",size=7,
                     vjust=.15,tip.length= c(0,0,0)) + 
  labs(x=NULL,y='Oxygenation index',)  +
  theme_pubr(base_size = 21,legend ='none',x.text.angle = 30) 

df_tempral_dyanmics_group_new_criteria_neutrophil <- df_tempral_dyanmics_group_new_criteria[!is.na(df_tempral_dyanmics_group_new_criteria$The.Percentage.of.Neutrophils.on.The.Day.of.Sampling), ]
ggboxplot(df_tempral_dyanmics_group_new_criteria_neutrophil, x="Group", y="The.Percentage.of.Neutrophils.on.The.Day.of.Sampling",
          color = "Group",palette =c("#E8080A", "#0A74B2"), 
          add = "jitter",add.params = list(size =2.4, alpha = 0.8))  +
  geom_boxplot(aes(color=Group), lwd = 0.7) +
  scale_x_discrete(labels = c("multiple" = "Multi-dominant", "stable" = "Mono-dominant")) +
  stat_compare_means(comparisons = my_comparisons,label = "p.label",size=7,
                     vjust=.15,tip.length= c(0,0,0)) + 
  labs(x=NULL,y='Percentage of neutrophils (%)',) +  
  theme_pubr(base_size = 21,legend ='none',x.text.angle = 30)

```

```{r Fig 3a}

df_arg_num = read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Main_figures\RGI_profile-comparisons_arg_number.csv)")
df_arg_num$species <- factor(df_arg_num$species,levels = c('pseudomonas','acinetobacter','klebsiella','corynebacterium','stenotrophomonas'))
my_comparisons <- list(c('pseudomonas','stenotrophomonas'),c('pseudomonas','corynebacterium'),
                       c('acinetobacter','stenotrophomonas'),c('acinetobacter','corynebacterium'),
                       c('klebsiella','stenotrophomonas'),c('klebsiella','corynebacterium'))

map <-c('acinetobacter'=3.965,'pseudomonas'=6.644,'klebsiella'=5.596,'corynebacterium'=2.841,'stenotrophomonas'=4.613)
df_arg_num <- df_arg_num %>%
  mutate(genome_length = map[species])
df_arg_num <- df_arg_num %>%
  mutate(arg_num_normalized = arg_num/genome_length)

ggboxplot(df_arg_num, x="species", y="arg_num_normalized",
          color = "species", palette =c("#E41A1C","#377EB8","#4DAF4A","#984EA3","#FF7F00","#F781BF","#A65628","#F781BF"),
          add = "jitter",add.params = list(size = 2, alpha = 0.5))  +
  stat_boxplot(mapping = aes(color=species),geom = "errorbar",width=0.1,lwd=0.8) +
  stat_compare_means(comparisons = rev(my_comparisons),label = "p.label",size=10,
                     vjust=.15,tip.length= c(0,0,0)) +
  scale_x_discrete(labels=c('P. aeruginosa','A. baumannii','K. pneumoniae','C. striatum','S. maltophilia'))+
  labs(x=NULL,y='Number of ARGs (Normalized)',) +  
  theme_pubr(base_size = 32,legend ='none',x.text.angle = 30) 


```

```{r Fig 3b}
df_meta_argname <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Main_figures\RGI_profile-species_all-metadata-arg_acronym.csv)')

dend_species_all <- ReadDendrogram(r"(LRT_microbiomes_critical_illness\Data_repository\Main_figures\inputfiles.tre-test.nwk)")
tree_species_all <- read.tree(r"(LRT_microbiomes_critical_illness\Data_repository\Main_figures\inputfiles.tre.nwk)")
order_row <- data.frame(species=tree_species_all$tip.label)


plot_heatmap_species_all <- Heatmap(t(matrix_rgi_species_all),
                                    #breaks = c(0,1,3),
                                    col = c('#ffffff','#ff0000'),
                                    # col = c('#f7faff','#08306b'),
                                    left_annotation  = rowAnnotation(df=labels_annotation_col,
                                                                     col=list(species=c(corynebacterium="#e36255", stenotrophomonas="#ec9a86",
                                                                                        pseudomonas="#a2c5c9", klebsiella="#f3c262", 
                                                                                        acinetobacter="#f1e0ce"))),
                                    #                                     col = list(category=c(core='#16519f',accessory='#f07e74'))),
                                    # row_split = factor(labels_annotation_col[,1],levels = c("corynebacterium",'stenotrophomonas','klebsiella','pseudomonas',"acinetobacter")),
                                    # row_split = c(30,9,12,13,27),
                                    #border_color = NA,
                                    # cluster_rows =as.dendrogram(dendrogram),
                                    # cluster_rows = F,
                                    cluster_columns =T,
                                    # cluster_rows = F,cluster_columns =T,
                                    show_column_dend = F,show_row_dend = T,rect_gp = gpar(col= "grey"),
                                    show_column_names = T,show_row_names = T,
                                    row_order = order_row$species)

