## Libraries

library(ggVennDiagram)
library(VennDiagram)
library(ggplot2)
library(ggpubr)
library(ggvenn)
library(dplyr)
library(patchwork)

## Input data
# input data was stored at the directory 'Data_repository'

### Figure 1
### --------------------------------------------------------------------------------------------------------------------------------------------

```{r Fig 1b}

df_metaphlan_all <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Main_figures\microbiome_species-top_prevalent.csv)')
df_metaphlan_all[df_metaphlan_all$SamplingSite=='HospitalA',]$Name
df_input_list <- list(
  "Hospital A" = df_metaphlan_all[df_metaphlan_all$SamplingSite=='Hospital-A',]$taxonomy,
  "Hospital B" = df_metaphlan_all[df_metaphlan_all$SamplingSite=='Hospital-B',]$taxonomy,
  "Hospital C" = df_metaphlan_all[df_metaphlan_all$SamplingSite=='Hospital-C',]$taxonomy
)

venn.diagram(
  x = df_input_list,
  category.names = c("Hospital A" , "Hospital B" , "Hospital C"),
  filename = r'()', # fill the filepath where you want to save the plot
  output=TRUE,
  
  # Output features
  imagetype="png" ,
  height = 1500 , 
  width = 1500 , 
  resolution = 900,
  compression = "lzw",
  
  # Circles
  lwd = 0.7,
  # col=c("#440154ff", '#509fd2', '#5da94e'),
  col=c("#d93526", "#f7a902", "#1a6196"),
  # fill = c(alpha("#440154ff",0.4), alpha('#509fd2',0.4), alpha('#5da94e',0.4)),
  fill = c(alpha("#d93526",0.3), alpha('#f7a902',0.3), alpha('#1a6196',0.3)),
  cex = 0.7,
  fontfamily = "sans",
  cat.cex = 0.6,
  cat.default.pos = "outer",
  cat.pos = c(-27, 27, 135),
  cat.dist = c(0.055, 0.055, 0.085),
  cat.fontfamily = "sans",
  # cat.col = c("#440154ff", '#509fd2', '#5da94e'),
  cat.col = c("#d93526", "#f7a902", "#1a6196"),
  rotation = 1
)

```

```{r Fig 1c}

df_prevalent <- read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Main_figures\microbiome_species-top_prevalent.csv)")
df_order <- read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Main_figures\microbiome_species-top_prevalent-orders.csv)")
df_sorted_desc <- arrange(df_order, desc(prevalency_percentage_all))

order_species <- df_sorted_desc[1:10,'taxonomy']
df_prevalent$taxonomy <- factor(df_prevalent$taxonomy, levels = df_sorted_desc[1:10,'taxonomy'])

#transform species name into human-readable format
df_prevalent$taxonomy <- gsub("^s__", "", df_prevalent$taxonomy) # Remove the prefix
df_prevalent$taxonomy <- gsub("_", " ", df_prevalent$taxonomy)
plot_df <- ggplot(na.omit(df_prevalent), aes(fill=SamplingSite, x=reorder(taxonomy,prevalency_percentage,sum,decreasing=T), y=prevalency_percentage))+  #'x=reorder(species,prevalency_percentage,sum,decreasing=T)' --> order stacked barplot with sum of values
  geom_bar(position = "dodge",stat = "identity",) +
  scale_fill_manual(values = c('Hospital-A'='#d93526','Hospital-B'='#f7a902','Hospital-C'='#1a6196'))+
  theme_pubr(x.text.angle = 35)

plot_df
ggpar(plot_df,
      ylab = "Percentage of samples with\ndetected species (%)",
      xlab = "Species",
      font.xtickslab=c(18),
      font.ytickslab=c(18),
      font.x = c(18,"bold"),
      font.y = c(18,"bold"),
      font.legend = c(18,'bold'),
      x.text.angle =50) + theme_pubr(base_size = 24,x.text.angle =35,legend = "none",margin = FALSE)

```

```{r Fig 1d}

tree_phylum_level2 <- read.tree(r'(LRT_microbiomes_critical_illness\Data_repository\Main_figures\gtdbtk_filtered_satisfied.tree)')
labels_tree_phylogeney.df  <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Main_figures\ggtree_assembled_bins_metadata_with_hospitals.csv)')
meta_info <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Main_figures\metadata-assembled_bins.csv)')


### plot the phylogenetic tree and the strip layer(display information about phylum level taxonomy)
p_final_layer1 <- ggtree(tree_phylum_level2, branch.length="none",layout = 'fan',open.angle=18,size=0.3) %<+% labels_tree_phylogeney.df + 
  #Proteobacateria
  geom_strip(taxa1 = "P139-HospitalA_coassembly.001",
             taxa2 = "P122-HospitalA_coassembly.002",
             offset = 8.8,
             barsize = 40,
             extend = 0.5,
             color="#6d9fcd") + #geom_tiplab(size=1.5,align = TRUE,offset = 0.5) + 
  #Acinetobacteria
  geom_strip(taxa1 = "P34-HospitalB_coassembly.037",
             taxa2 = "P38-HospitalB_coassembly.004",
             offset = 8.8,
             barsize = 40,
             extend = 0.5,
             color="#c3e5f5") +
  #Firmicutes
  geom_strip(taxa1 = "P34-HospitalB_coassembly.038",
             taxa2 = "P123-HospitalA_coassembly.005",
             offset = 8.8,
             barsize = 40,
             extend = 0.5,
             color="#A4A7DE") +
  #Firmicutes
  geom_strip(taxa1 = "P34-HospitalB_coassembly.022",
             taxa2 = "P123-HospitalA_coassembly.007",
             offset = 8.8,
             barsize = 40,
             extend = 0.5,
             color="#A4A7DE") +
  #Firmicutes
  geom_strip(taxa1 = "P34-HospitalB_coassembly.027",
             taxa2 = "P34-HospitalB_coassembly.050",
             offset = 8.8,
             barsize = 40,
             extend = 0.5,
             color="#A4A7DE") +
  #Firmicutes
  geom_strip(taxa1 = "P45-HospitalB_coassembly.008",
             taxa2 = "P22-HospitalB_coassembly.002",
             offset = 8.8,
             barsize = 40,
             extend = 0.5,
             color="#A4A7DE") +
  #Bacteroidota
  geom_strip(taxa1 = "P72-HospitalB_coassembly.005",
             taxa2 = "P44-HospitalB_coassembly.009",
             offset = 8.8,
             barsize = 40,
             extend = 0.5,
             color="#ffc3c3") + 
  #Actinobacteriota
  geom_strip(taxa1 = "P34-HospitalB_coassembly.014",
             taxa2 = "P24-HospitalB_coassembly.005",
             offset = 8.8,
             barsize = 40,
             extend = 0.5,
             color="#c3e5f5")


## show collection sites with various star types 
p_final_layer2 <- p_final_layer1 +
  geom_fruit(data=meta_info, geom=geom_star,
             mapping=aes(y=user_genome,x=sampling_site,starshape=sampling_site,fill=sampling_site),
             color = "black", offset=0.30,
             size =2,
             pwidth=0.1,
             grid.params=list(
               linetype=2,
               size=0.5
             )) + scale_fill_manual(values=c("#FFC125","#87CEFA","#7B68EE"),
                                    guide=guide_legend(override.aes = list(size = 8)))  + 
  geom_tiplab(aes(label=species),size=1.7,offset = 1.2) + new_scale_fill()  



## show collections sites with various colors
# p_final_layer2 <- p_final_layer1 +
#   geom_fruit(data=meta_info, geom=geom_tile,
#              mapping=aes(y=user_genome,fill=sampling_site),
#              color = "black", offset = 0.5,size = 0.02,width=2.4,) +
#   scale_fill_manual(values=c("#FFC125","#87CEFA","#7B68EE")) + geom_tiplab(size=1.5)
#   
p_final_layer3 <- p_final_layer2 + 
  geom_fruit(data=meta_info, geom=geom_bar,
             mapping=aes(y=user_genome, x=fastani_af, fill=sampling_site),
             offset = 0.05,
             pwidth=0.1,
             width=0.6,
             orientation="y", 
             stat="identity",
             show.legend=F,
             axis.params = list(axis='x',text.size=2,nbreak=2,vjust=1)) +
  scale_fill_manual(values=c("#FFC125","#87CEFA","#7B68EE"))  + new_scale_fill() +
  geom_tippoint(mapping=aes(color=phylum),
                size=1.2,
                show.legend=F) +
  scale_color_manual(values=c("Proteobacteria"="#6d9fcd",
                              "Actinobacteriota"="#c3e5f5",
                              "Firmicutes"="#A4A7DE",
                              "Firmicutes_A"="#A4A7DE",
                              "Firmicutes_C"="#A4A7DE",
                              "Bacteroidota"="#ffc3c3")) 


p_final_layer4 <- p_final_layer3 + 
  # Acinetobacter baumannii
  geom_strip(taxa1 = "P123-HospitalA_coassembly.002",
             taxa2 = "P122-HospitalA_coassembly.002",
             offset = 29,
             barsize =0.8,
             extend = 0.5,
             color="#267278",
             alpha=1) +
  # Burkholderia cenocepacia
  geom_strip(taxa1 = "P157-HospitalA_coassembly.003",
             taxa2 = "P66-HospitalB_coassembly.002",
             offset = 29,
             barsize =0.8,
             extend = 0.5,
             color="#65338D",
             alpha=1) +
  geom_strip(taxa1 = "P14-HospitalC_coassembly.002",
             taxa2 = "P6-HospitalC_coassembly.002",
             offset = 29,
             barsize =0.8,
             extend = 0.5,
             color="#65338D",
             alpha=1) +
  # Corynebacterium striatum
  geom_strip(taxa1 = "P121-HospitalA_coassembly.003",
             taxa2 = "P24-HospitalB_coassembly.005",
             offset = 29,
             barsize =0.8,
             extend = 0.5,
             color="#4770B3",
             alpha=1) +
  geom_strip(taxa1 = "P155-HospitalA_coassembly.004",
             taxa2 = "P155-HospitalA_coassembly.004",
             offset = 29,
             barsize =0.8,
             extend = 0.5,
             color="#4770B3",
             alpha=1) +
  # Elizabethkingia anophelis
  geom_strip(taxa1 = "P141-HospitalA_coassembly.002",
             taxa2 = "P141-HospitalA_coassembly.001",
             offset = 29,
             barsize =0.8,
             extend = 0.5,
             color="#D21F75",
             alpha=1) +
  # Enterococcus_B faecium
  geom_strip(taxa1 = "P96-HospitalB_coassembly.002",
             taxa2 = "P76-HospitalB_coassembly.001",
             offset = 29,
             barsize =0.8,
             extend = 0.5,
             color="#3B3689",
             alpha=1) +
  # Klebsiella pneumoniae
  geom_strip(taxa1 = "P116-HospitalA_coassembly.002",
             taxa2 = "P155-HospitalA_coassembly.002",
             offset = 29,
             barsize =0.8,
             extend = 0.5,
             color="#50AED3",
             alpha=1) +
  geom_strip(taxa1 = "P115-HospitalA_coassembly.001",
             taxa2 = "P156-HospitalA_coassembly.002",
             offset = 29,
             barsize =0.8,
             extend = 0.5,
             color="#50AED3",
             alpha=1)+
  geom_strip(taxa1 = "P96-HospitalB_coassembly.001",
             taxa2 = "P96-HospitalB_coassembly.001",
             offset = 29,
             barsize =0.8,
             extend = 0.5,
             color="#50AED3",
             alpha=1) +
  # Porphyromonas pasteri
  geom_strip(taxa1 = "P121-HospitalA_coassembly.002",
             taxa2 = "P32-HospitalB_coassembly.008",
             offset = 29,
             barsize =0.8,
             extend = 0.5,
             color="#48B24F",
             alpha=1) +
  # Prevotella melaninogenica
  geom_strip(taxa1 = "P23-HospitalB_coassembly.006",
             taxa2 = "P45-HospitalB_coassembly.006",
             offset = 29,
             barsize =0.8,
             extend = 0.5,
             color="#E57438",
             alpha=1) +
  geom_strip(taxa1 = "P50-HospitalB_coassembly.004",
             taxa2 = "P34-HospitalB_coassembly.003",
             offset = 29,
             barsize =0.8,
             extend = 0.5,
             color="#E57438",
             alpha=1) +
  # Prevotella oris
  geom_strip(taxa1 = "P72-HospitalB_coassembly.013",
             taxa2 = "P47-HospitalB_coassembly.010",
             offset = 29,
             barsize =0.8,
             extend = 0.5,
             color="#569DD2",
             alpha=1) +
  # Pseudomonas aeruginosa
  geom_strip(taxa1 = "P20-HospitalC_coassembly.001",
             taxa2 = "P152-HospitalA_coassembly.004",
             offset = 29,
             barsize =0.8,
             extend = 0.5,
             color="#569D79",
             alpha=1) +
  # Stenotrophomonas maltophilia
  geom_strip(taxa1 = "P105-HospitalB_coassembly.003",
             taxa2 = "P139-HospitalA_coassembly.001",
             offset = 29,
             barsize =0.8,
             extend = 0.5,
             color="#58595B",
             alpha=1) + 
  # Haemophilus influenzae
  geom_strip(taxa1 = "P54-HospitalB_coassembly.001",
             taxa2 = "P44-HospitalB_coassembly.002",
             offset = 29,
             barsize =0.8,
             extend = 0.5,
             color="black",
             alpha=1) + 
  # Slackia exigua
  geom_strip(taxa1 = "P45-HospitalB_coassembly.020",
             taxa2 = "P72-HospitalB_coassembly.026",
             offset = 29,
             barsize =0.8,
             extend = 0.5,
             color="black",
             alpha=1) + 
  # Staphylococcus aureus
  geom_strip(taxa1 = "P34-HospitalB_coassembly.029",
             taxa2 = "P155-HospitalA_coassembly.010",
             offset = 29,
             barsize =0.8,
             extend = 0.5,
             color="black",
             alpha=1) + 
  # Streptococcus anginosus
  geom_strip(taxa1 = "P80-HospitalB_coassembly.001",
             taxa2 = "P80-HospitalB_coassembly.001",
             offset = 29,
             barsize =0.8,
             extend = 0.5,
             color="black",
             alpha=1) +
  geom_strip(taxa1 = "P112-HospitalA_coassembly.001",
             taxa2 = "P112-HospitalA_coassembly.001",
             offset = 29,
             barsize =0.8,
             extend = 0.5,
             color="black",
             alpha=1) +
  geom_strip(taxa1 = "P47-HospitalB_coassembly.004",
             taxa2 = "P47-HospitalB_coassembly.004",
             offset = 29,
             barsize =0.8,
             extend = 0.5,
             color="black",
             alpha=1) +
  # Streptococcus constellatus
  geom_strip(taxa1 = "P45-HospitalB_coassembly.009",
             taxa2 = "P50-HospitalB_coassembly.005",
             offset = 29,
             barsize =0.8,
             extend = 0.5,
             color="black",
             alpha=1) +
  geom_strip(taxa1 = "P102-HospitalB_coassembly.006",
             taxa2 = "P102-HospitalB_coassembly.006",
             offset = 29,
             barsize =0.8,
             extend = 0.5,
             color="black",
             alpha=1) 

```

```{r Fig 1e}

```
### Figure S1
### --------------------------------------------------------------------------------------------------------------------------------------------

```{r Fig S1a}

#most prevalent species for hospitalA
df_species_hospitalA <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\microbiome_species-top_prevalent.csv\microbiome_species-statistics_count-Hospital-A.csv)',header = TRUE) # save in the same file 'microbiome_species-statistics_count-Hospitals.txt'
df_species_hospitalA$taxonomy <- gsub("^s__", "", df_species_hospitalA$taxonomy) # Remove the prefix
df_species_hospitalA$taxonomy <- gsub("_", " ", df_species_hospitalA$taxonomy)

plot_species_hospitalA <- ggbarplot(df_species_hospitalA[1:5,],x = "taxonomy", y = "prevalency_percentage",orientation = "vertical",alpha=1,
                                    fill = "#d93526", color = "#d93526",sort.val = 'desc',ggtheme = theme_bw(base_size = 18)) + 
  labs(x=NULL,y='Percentage of\ndetected samples (%)') +  theme(aspect.ratio =1/1.5 )  #+ #+ ylim(0,15) 
  #scale_y_continuous(breaks = c(0,2,4,6,8,10))

plot_species_hospitalA <-ggpar(plot_species_hospitalA,x.text.angle = 55)

#-------------------------------------------------------------------------------------------------------------------------------

#most prevalent species for hospitalB
df_species_hospitalB <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\microbiome_species-top_prevalent.csv\microbiome_species-statistics_count-Hospital-B.csv)',header = TRUE # save in the same file 'microbiome_species-statistics_count-Hospitals.txt'
df_species_hospitalB$taxonomy <- gsub("^s__", "", df_species_hospitalB$taxonomy) # Remove the prefix
df_species_hospitalB$taxonomy <- gsub("_", " ", df_species_hospitalB$taxonomy)

plot_species_hospitalB <- ggbarplot(df_species_hospitalB[1:5,],x = "taxonomy", y = "prevalency_percentage",orientation = "vertical",alpha=1,
                                    fill = "#f7a902", color = "#f7a902",sort.val = 'desc',ggtheme = theme_bw(base_size = 18)) + 
  labs(x='Species')  +  theme(aspect.ratio =1/1.5,axis.title.y = element_blank())

plot_species_hospitalB <- ggpar(plot_species_hospitalB,x.text.angle = 55)

#-------------------------------------------------------------------------------------------------------------------------------

#most prevalent species for hospitalC
df_species_hospitalC <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\microbiome_species-top_prevalent.csv\microbiome_species-statistics_count-Hospital-C.csv)',header = TRUE) # save in the same file 'microbiome_species-statistics_count-Hospitals.txt'
df_species_hospitalC$taxonomy <- gsub("^s__", "", df_species_hospitalC$taxonomy) # Remove the prefix
df_species_hospitalC$taxonomy <- gsub("_", " ", df_species_hospitalC$taxonomy)
df_species_hospitalC <- df_species_hospitalC[-5,]

plot_species_hospitalC <- ggbarplot(df_species_hospitalC[1:5,],x = "taxonomy", y = "prevalency_percentage",orientation = "vertical",alpha=1,
                                    fill = "#1a6196", color = "#1a6196",sort.val = 'desc',ggtheme = theme_bw(base_size = 18)) + 
  labs(x=NULL)  +  theme(aspect.ratio =1/1.5,axis.title.y = element_blank() )

plot_species_hospitalC <-ggpar(plot_species_hospitalC,x.text.angle = 55)

plot_species_hospitalA|plot_species_hospitalB|plot_species_hospitalC

```

```{r Fig S1b}

df_too_low <- read.csv((r'(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\microbiome_species-top_prevalent.csv\head-to-head_comparisons-DNA_concentration_too_low.csv)')) # save in the same file 'head-to-head_comparisons-DNA_concentration.txt'
df_too_low$Percentage_Failed <- df_too_low$Percentage_Failed *100
df_too_low$Percentage_successful <- 100 - df_too_low$Percentage_Failed
plot <- ggbarplot(df_too_low, "Method", "Percentage_successful",
                  color = '#1f78b4',fill='#1f78b4',palette = "#1f78b4",
                  label = F,
                  position = position_dodge(0.9)) + 
  scale_x_discrete(labels= c('CMEM','Power Water','AllPrep DNA')) + 
  labs(x=NULL,y='Percentage of samples with detectable\nDNA after extraction (%)') + theme_bw(base_size = 27)
ggpar(plot,legend = 'bottom',x.text.angle = 30)

```

```{r Fig S1c}

df_concentration <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\head-to-head_comparisons-DNA_concentration.csv)')
df_concentration$Concentration  <- log10(df_concentration$Concentration + 1)
df_concentration$Method <- factor(df_concentration$Method,levels =c('CMEM','Power Water','AllPrep DNA') )
my_comparisons <- list(c('CMEM','Power Water'),c('CMEM','AllPrep DNA'),c('Power Water','AllPrep DNA'))
p_concentration <- ggboxplot(df_concentration, "Method", "Concentration",
          color = "Method", palette =c("#00AFBB", "#E7B800", "#FC4E07"),
          add = "jitter",add.params = list(size = 1.5, alpha = 0.5))  +
  stat_boxplot(geom = "errorbar",width=0.05,color=c("#00AFBB", "#E7B800", "#FC4E07"))  +
  stat_compare_means(comparisons = my_comparisons,label = "p.label",size=8,vjust=0) +
  scale_x_discrete(labels= c('CMEM','Power Water','AllPrep DNA')) + 
  labs(x=NULL,y='DNA concentration (log10, ng/ul)',) + #ylim(-1,10) +
  theme_bw(base_size = 24)

```

```{r Fig S1d}

df_mappingrate <- read.csv(r'(G:\Bio-Analysis\Python\Vscode\Task01-NEW\Responses-related\files-reformat\MappingRate_ETAs.csv)')
df_mappingrate$microbial <- 100 - df_mappingrate$HumanMappingRate
df_mappingrate$fold_increase <- 100 - df_mappingrate$HumanMappingRate -1

plot_microbial_reads <- ggboxplot(df_mappingrate, x=NULL, y="microbial",
                              color = "#2077b4", palette =c("#2077b4", "#E7B800", "#FC4E07"),
                              add = "jitter",add.params = list(size = 1.5, alpha = 0.5)) +
  stat_boxplot(geom = "errorbar",width=0.05,color='#2077b4')  +
  scale_x_discrete(labels='Samples') +
  labs(x=NULL,y='Proportion of microbial reads (%)') + theme_bw(base_size = 16)+
  theme(axis.text.x = element_text(angle=30,hjust=0.7,vjust=0.8)) 

```

```{r Fig S1e}

df_profiles_vegan  <-  read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\microbial_profile-feasibility_validation.csv)",header = TRUE) # save in the same file 'microbial_profile-feasibility_validation.txt'
metadata = read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\metadata-microbial_profile-feasibility_validation.csv)",header = TRUE)  # save in the same file 'microbial_profile-feasibility_validation.txt'

df_profiles_vegan_abundance <- df_profiles_vegan[,3:ncol(df_profiles_vegan)]
df_profiles_vegan_matrix <- as.matrix(df_profiles_vegan[,3:ncol(df_profiles_vegan)])

abundance <- select(df_profiles_vegan,-c(SampleID,Group))
rownames(abundance) <- df_profiles_vegan[1:nrow(df_profiles_vegan),1]
abundance <- t(abundance) #transposed dataframe

pca_metadata <- data.frame(df_profiles_vegan[1:nrow(df_profiles_vegan),2])
rownames(pca_metadata) <- df_profiles_vegan[1:nrow(df_profiles_vegan),1]
colnames(pca_metadata) <- c('Group')
p <- pca(abundance,metadata=pca_metadata)

# adonis test 
set.seed(123)
adonis2(df_profiles_vegan_matrix ~ Group, df_profiles_vegan, method = "bray", permutations = 999)

df_pca_output <- as.data.frame(p$rotated)
df_pca_output$SampleID <- rownames(df_pca_output)
df_pca_output <- merge(df_pca_output, metadata, by = "SampleID")

ggscatter(df_pca_output, x = "PC1", y = "PC2", color = "Group", palette = c("#0466AB", "#D15D4E"), 
           size =5, alpha = 1) +
  geom_line(data = df_pca_output, aes(group = Order), color = "black",alpha = 0.8) +
  # stat_ellipse(aes(color = Group), geom = "path", alpha = 1, linetype = "solid",linewidth =0.8,
  #              level = 0.9) + theme_pubr(border = T)
  stat_ellipse(aes(fill = Group), geom = "polygon", linetype = "solid", level = 0.95,alpha=0.3) +
#   scale_fill_manual(values = c("#0466AB", "#FFEFF2"))
  theme_pubr(base_size = 23,border = T)

```

```{r Fig S1f}

df_diversity  <- read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\alpha_diversity-feasibility_validation-comparison.csv)",header = TRUE)
df_diversity$Group  <- factor(df_diversity$Group,levels = c("Undepleted","Depleted"))


my_comparisons <- list(c("Undepleted","Depleted"))

p_diversity_shannon <- ggplot(df_diversity, aes(x=Group,y=Alpha_diversity_Shannon,fill=Group)) +
  stat_boxplot(geom = "errorbar",width=0.15)+
  geom_boxplot(size=0.5 ) + #theme_classic2() +
  geom_line(aes(group=Order),colour='Gray75', size=0.7)+
  geom_line(size=0.3) +
  scale_fill_manual(values = c("#00AFBB", "#E7B800")) +
  scale_x_discrete(labels = c("Control","Treatment")) +
  stat_compare_means(paired = T,label.y = 2.2,comparisons = my_comparisons,label = "p.signif",size=7,bracket.size = 0.5) +
  stat_compare_means(paired = T,label.y = 2.5,label.x = 1,size=8) + 
  labs(x = NULL, y = "Alpha diversity (Shannon)") +
  theme_pubr(base_size = 20, border =F,legend = "none")

p_diversity_simpson <- ggplot(df_diversity, aes(x=Group,y=Alpha_diversity_Simpson,fill=Group)) +
  stat_boxplot(geom = "errorbar",width=0.15)+
  geom_boxplot(size=0.5 ) + #theme_classic2() +
  geom_line(aes(group=Order),colour='Gray75', size=0.7)+
  geom_line(size=0.3) +
  scale_fill_manual(values = c("#00AFBB", "#E7B800")) +
  scale_x_discrete(labels = c("Control","Treatment")) +
  stat_compare_means(paired = T,label.y = 0.9,comparisons = my_comparisons,label = "p.signif",size=7,bracket.size = 0.5) +
  stat_compare_means(paired = T,label.y = 1.1,label.x = 1,size=8) + 
  labs(x = NULL, y = "Alpha diversity (Simpson)") +
  theme_pubr(base_size = 20, border =F,legend = "none")

```

```{r Fig S1g}

df_total_args  <- read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\total_arg_abundance-feasibility_validation.csv)",header = TRUE)
df_total_args$Group  <- factor(df_total_args$Group,levels = c("Undepleted","Depleted"))
df_total_args$total_arg_abundance_log10 <- log10(df_total_args$total_arg_abundance)
my_comparisons <- list(c("Undepleted","Depleted"))
p_arg_abundance <- ggplot(df_total_args, aes(x=Group,y=total_arg_abundance_log10,fill=Group)) +
  stat_boxplot(geom = "errorbar",width=0.15)+
  geom_boxplot(size=0.5 ) + #theme_classic2() +
  geom_line(aes(group=Order),colour='Gray75', size=0.7)+
  geom_line(size=0.3) +
  scale_fill_manual(values = c("#00AFBB", "#E7B800")) +
  scale_x_discrete(labels = c("Control","Treatment")) +
  stat_compare_means(paired = T,label.y = 1.8,comparisons = my_comparisons,label = "p.signif",size=7,bracket.size = 0.5) +
  stat_compare_means(paired = T,label.y = 2.5,label.x = 1,size=8) + 
  labs(x = NULL, y = "Total ARG abundance (log10)") +
  theme_pubr(base_size = 20, border =F,legend = "none")

```

```{r Fig S1h}

df_diversity  <- read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\alpha_diversity-comparison-pneumonia_vs_no_pneumonia.csv)",header = TRUE)
df_merged_subset$Diagnosed.Pneumonia   <- factor(df_merged_subset$Diagnosed.Pneumonia,levels = c("No","Yes"))

my_comparisons <- list(c("No",'Yes'))

p_diversity_shannon <- ggplot(df_merged_subset, aes(x=Diagnosed.Pneumonia,y=Alpha_diversity_Shannon,fill=Diagnosed.Pneumonia)) +
  stat_boxplot(geom = "errorbar",width=0.15)+
  geom_boxplot(size=0.5 ) + #theme_classic2() +
#   geom_line(aes(group=Order),colour='Gray75', size=0.7)+
  geom_line(size=0.3) +
  scale_fill_manual(values = c("#00AFBB", "#E7B800")) +
  scale_x_discrete(labels = c("No Pneumonia","Pneumonia")) +
  stat_compare_means(label.y = 3.2,comparisons = my_comparisons,label = "p.signif",size=7,bracket.size = 0.5) +
  stat_compare_means(label.y = 3.8,label.x = 1,size=8) + 
  labs(x = NULL, y = "Alpha diversity (Shannon)") +
  theme_pubr(base_size = 21, border =F,legend = "none")

ggpar(p_diversity_shannon,x.text.angle = 30,legend = 'none')

```

```{r Fig S1i}

df_arg_abundance  <- read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\argtypes-comparison-pneumonia_vs_nopneumonia.csv)",header = TRUE)
df_arg_abundance$totoal_arg_abundance_log10 <- log10(df_arg_abundance$totoal_arg_abundance)
my_comparisons <- list(c("No",'Yes'))
p_arg_abundance <- ggplot(df_arg_abundance, aes(x=Diagnosed.Pneumonia,y=totoal_arg_abundance_log10,fill=Diagnosed.Pneumonia)) +
  stat_boxplot(geom = "errorbar",width=0.15)+
  geom_boxplot(size=0.5 ) + #theme_classic2() +
  geom_line(size=0.3) +
  scale_fill_manual(values = c("#00AFBB", "#E7B800")) +
  scale_x_discrete(labels = c("No Pneumonia","Pneumonia")) +
  stat_compare_means(label.y = 3.0,comparisons = my_comparisons,label = "p.signif",size=7,bracket.size = 0.5) +
  stat_compare_means(label.y = 3.8,label.x = 1,size=8) + 
  labs(x = NULL, y = "Total ARG abundance (log10)") +
  theme_pubr(base_size = 20, border =F,legend = "none")
ggpar(p_arg_abundance,x.text.angle = 30,legend = 'none')

```

### Figure 2
### --------------------------------------------------------------------------------------------------------------------------------------------

```{r Fig 2a}

df_distance_bray <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Main_figures\MDS_metrics-bray-species_profiles_pneumonia.csv)')  # save in the same file 'MDS_metrics-bray-species_profiles_pneumonia.txt'
df_arrows <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Main_figures\MDS_metrics-bray-specie_pneumonia-arrows_coordinates.csv)') # save in the same file 'MDS_metrics-bray-species_profiles_pneumonia.txt'
df_arrows$variable <- str_split(df_arrows$variable, "__", simplify = TRUE)[,2]
df_arrows$variable  <- str_replace_all(df_arrows$variable, "_", " ")
df_arrows$variable  <- str_replace_all(df_arrows$variable, " group", " ")

#remove plot direction
df_distance_bray$Axis.2 <- df_distance_bray$Axis.2*-1
df_arrows$Axis.2 <- df_arrows$Axis.2*-1

plots <- ggscatter(df_distance_bray,
                   x = "Axis.1", 
                   y = "Axis.2",
                   color = "SamplingSite",
                   #shape = "SamplingSite",
                   size=6,
                   # palette = c( "#FC4E07","#00AFBB", "#E7B800"), #color1
                   palette = c("#e64833", "#01a188", "#3e5386"), #color2
                   # palette = c("#c23726", "#e3e3e3", "#0036fc"),  #palette2
                   # palette = c( "#010101","#fe0000", "#0037fe"), #palette4
                   #palette = c("#c23726", "#5a5a5a", "#2c83c6"),
                   #palette = c("#7152a9", "#ffc376", "#eb7600"),
                   margin.plot = "boxplot",
                   ggtheme = theme_pubr(),
                   #margin.params = list(fill="SamplingSite",size=0.36),
                   legend = 'right') + 
  geom_segment(data = df_arrows,
               x = 0, y = 0, alpha = 0.7,
               mapping = aes(xend = Axis.1, yend = Axis.2),
               color = 'black',size=0.1,
               # Add arrow head
               arrow = arrow(length = unit(3, "mm"))) +
  geom_label(data = df_arrows, aes(label = variable),color='black',alpha=0.7,size=7,label.size =0.01) 


ggpar(plots,
      xlab = "MDS1 (27.6%)",
      ylab = "MDS2 (16.6%)") + theme_pubr(base_size = 24,border = T,legend = 'none')



```

```{r Fig 2b}

df_distance_bray_species_pneumonia <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Main_figures\MDS_metrics-bray+species_abundance-species_pneumonia.csv)') # save in the same file 'MDS_metrics-bray-species_profiles_pneumonia.txt'
df_distance_bray_species_pneumonia$variable <- str_split(df_distance_bray_species_pneumonia$variable, "__", simplify = TRUE)[,2]
df_distance_bray_species_pneumonia$variable  <- str_replace_all(df_distance_bray_species_pneumonia$variable, "_", " ")
df_distance_bray_species_pneumonia$variable  <- str_replace_all(df_distance_bray_species_pneumonia$variable, " group", " ")
df_distance_bray_species_pneumonia$Axis.2 <- df_distance_bray_species_pneumonia$Axis.2*-1

plots_mds_species <- ggscatter(df_distance_bray_species_pneumonia,
                   x = "Axis.1", 
                   y = "Axis.2",
                   # color = "Streptococcus",
                   color = 'value',
                   #shape = "SamplingSite",
                   size=6,
                   margin.plot = "boxplot",
                   ggtheme = theme_pubr(border = T),
                   #margin.params = list(fill="Acinetobacter",size=0.36),
                   legend = 'right') + #gradient_color(inferno(10))
                   gradient_color(c('#bfbdbd','#fe1608')) + facet_wrap(~ variable) +
  labs(x='MDS1 (19.8%)',y='MDS2 (14.5%)',color='Relative\nabundance (%)') + 
  theme(axis.text.x  = element_text(size=32),axis.text.y = element_text(size=32),
        axis.title.x = element_text(size = 32),axis.title.y = element_text(size = 32)) +
  theme(strip.background = element_blank(),
        strip.text = element_textbox(
          size = 30,
          color = "black", fill = "#d9d9d9", box.color = "#4A618C",#face = 'italic',
          halign = 0.5, linetype = 1,  width = unit(1, "npc"),
          margin = margin(0, 0, 0, 0),padding = margin(6, 6, 6, 6)),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.x = element_blank(),
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank(),
        legend.position = 'none',
        legend.title = element_text(size=14),
        legend.text = element_text(size=16)) +
  xlim(-0.5,0.53)
plots_mds_species

```

```{r Fig 2c,d}

patientids <- c('P68','P23')
df_abundance <- read.csv(paste("LRT_microbiomes_critical_illness\Data_repository\Main_figures",
                               patientid,"-metaphlan3-species_microbiota.csv",sep="")) # save in the same file 'pid_individual-metaphlan3-species_microbiota.txt'

df_abundance$Sampling_Day_ICU <- factor(df_abundance$Sampling_Day_ICU,levels = unique(df_abundance$Sampling_Day_ICU))
plot_df <- ggplot(df_abundance, aes(fill=taxonomy, x=Sampling_Day_ICU, y=relative_abundance))+
  geom_bar(position = "stack",stat = "identity",) + ggtitle(label = patientid ) + 
  scale_fill_manual(values = species2color)
ggpar(plot_df,
      ylab = "Relative\nabundance (%)",
      xlab = F) + theme_pubr(base_size = 14,x.text.angle =50,legend = "none",margin = FALSE)  + xlab(NULL) +
  ggeasy::easy_center_title() 

```

```{r Fig 2e}

### visualized with circos webserver

```

```{r Fig 2f}

df_argtypes_collection <- read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Main_figures\argtypes-collection_all_sites-pneumonia_patients.csv)")
df_argtypes_collection2 <- arrange(df_argtypes_collection,SamplingSite)
my_comparisons <- list( c("HospitalA", "HospitalB"), c("HospitalA", "HospitalC"), c("HospitalB", "HospitalC") )

#plot for 'multidrug'
ggboxplot(df_argtypes_collection2, x="SamplingSite", y="multidrug",
          color = "SamplingSite", palette =c("#f93800", "#007a8b","#ffb500"),
          add = "jitter",add.params = list(size = 2, alpha = 0.5))  +
  stat_boxplot(mapping = aes(color=SamplingSite),geom = "errorbar",width=0.1)  +
  geom_boxplot(aes(color=SamplingSite), lwd = 0.7,geom = "errorbar",) +
  stat_compare_means(comparisons = my_comparisons,label = "p.label",size=5,
                     vjust=.15,tip.length= c(0,0,0)) + 
  yscale("log10", .format = F)  + 
  labs(x=NULL,y='ARG abundance',) +  ggtitle('Multidrug') +
  theme_pubr(base_size = 18,legend ='none',x.text.angle = 30) +
  ggeasy::easy_center_title() 

#plot for 'aminoglycoside'
ggboxplot(df_argtypes_collection2, x="SamplingSite", y="aminoglycoside",
                             color = "SamplingSite", palette =c("#f93800", "#007a8b","#ffb500"),
                             add = "jitter",add.params = list(size = 2, alpha = 0.5))  +
  stat_boxplot(mapping = aes(color=SamplingSite),geom = "errorbar",width=0.1)  +
  geom_boxplot(aes(color=SamplingSite), lwd = 0.7,geom = "errorbar",) +
  stat_compare_means(comparisons = my_comparisons,label = "p.label",size=5,
                     vjust=.15,tip.length= c(0,0,0)) + 
  yscale("log10", .format = F)  + 
  labs(x=NULL,y='ARG abundance',) +  ggtitle('Aminoglycoside') +
  theme_pubr(base_size = 18,legend ='none',x.text.angle = 30) +
  ggeasy::easy_center_title() 


#plot for 'beta.lactam'
ggboxplot(df_argtypes_collection2, x="SamplingSite", y="beta.lactam",
          color = "SamplingSite", palette =c("#f93800", "#007a8b","#ffb500"),
          add = "jitter",add.params = list(size = 2, alpha = 0.5))  +
  stat_boxplot(mapping = aes(color=SamplingSite),geom = "errorbar",width=0.1)  +
  geom_boxplot(aes(color=SamplingSite), lwd = 0.7,geom = "errorbar",) +
  stat_compare_means(comparisons = my_comparisons,label = "p.label",size=5,
                     vjust=.15,tip.length= c(0,0,0)) + 
  yscale("log10", .format = F)  +
  labs(x=NULL,y='ARG abundance',) +  ggtitle('Beta-lactam') +
  theme_pubr(base_size = 18,legend ='none',x.text.angle = 30) +
  # theme(strip.text.x = element_text(size = 13.5)) +
  ggeasy::easy_center_title() 

```

```{r Fig 2g,h; Fig S3g}

df_combined  <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Main_figures\argsubtypes_pnuemonia_vs_nonpneumonia.csv)')
df_argtypes  <- df_combined %>%
  select(-c('Diagnosed.Pneumonia', 'Antibiotic.Use'))

# Get the names of the argtypes
argtypes <- colnames(df_argtypes)[!(colnames(df_argtypes) %in% c('SampleID', 'SamplingSite'))]
# Initialize a vector to store the p-values
p_values <- numeric(length(argtypes))
# Perform a Wilcoxon test for each argtype
for (i in seq_along(argtypes)) {
  pneumonia_group <- df_combined[df_combined$Diagnosed.Pneumonia == 'Yes', argtypes[i]]
  non_pneumonia_group <- df_combined[df_combined$Diagnosed.Pneumonia == 'No', argtypes[i]]
  wilcox_test_result <- wilcox.test(pneumonia_group, non_pneumonia_group)
  # Check if the p-value is less than 0.05 and the median of the pneumonia group is greater than the median of the non-pneumonia group
  if (wilcox_test_result$p.value < 0.05 && median(pneumonia_group, na.rm = TRUE) > median(non_pneumonia_group, na.rm = TRUE)) {
    p_values[i] <- wilcox_test_result$p.value
  } else {
    p_values[i] <- NA
  }
}

# Combine the argtypes and p-values into a data frame
df_p_values <- data.frame(Argtype = argtypes, P_Value = p_values)
df_p_values <- na.omit(df_p_values)

### plot based on the p-values for argtypes and argsubtypes
my_comparisons <- list(c("Yes", "No"))

df_combined$tetracycline__tetA_log10 <- log10(df_combined$tetracycline__tetA)
ggboxplot(df_combined, x="Diagnosed.Pneumonia", y="tetracycline__tetA_log10",
          color = "Diagnosed.Pneumonia", #palette =c("#0073C2", "#EFC000","#ffb500"),
        #   add = "dotplot",add.params = list(binwidth = 0.1, dotsize = 0.03))  +
          add = "jitter",add.params = list(size = 2.4, alpha = 0.8))  +
  geom_boxplot(aes(color=Diagnosed.Pneumonia), lwd = 0.7) +
  scale_x_discrete(labels = c("No" = "No Pneumonia", "Yes" = "Pneumonia")) +
  stat_compare_means(comparisons = my_comparisons,label = "p.label",size=7,
                     vjust=.15,tip.length= c(0,0,0)) + 
  labs(x=NULL,y='Total ARG abundance (log10)',) +  ggtitle('Tetracycline__tetA') +
  theme_pubr(base_size = 23,legend ='none',x.text.angle = 30) + 
  ggeasy::easy_center_title() + 
  scale_color_npg()


### ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
my_comparisons <- list(c("Yes", "No"))

df_combined$multidrug__mexB_log10 <- log10(df_combined$multidrug__mexB)
ggboxplot(df_combined, x="Diagnosed.Pneumonia", y="multidrug__mexB_log10",
          color = "Diagnosed.Pneumonia", #palette =c("#0073C2", "#EFC000","#ffb500"),
          #   add = "dotplot",add.params = list(binwidth = 0.1, dotsize = 0.03))  +
          add = "jitter",add.params = list(size = 2.4, alpha = 0.8))  +
  geom_boxplot(aes(color=Diagnosed.Pneumonia), lwd = 0.7) +
  # geom_boxplot(aes(color=Diagnosed.Pneumonia), lwd = 0.7) +
  scale_x_discrete(labels = c("No" = "No Pneumonia", "Yes" = "Pneumonia")) +
  #   stat_boxplot(mapping = aes(color=Diagnosed.Pneumonia),geom = "errorbar",width=0.05)  +
  stat_compare_means(comparisons = my_comparisons,label = "p.label",size=7,
                     vjust=.15,tip.length= c(0,0,0)) + 
  #   yscale("log10",.format = T)  +
  labs(x=NULL,y='Total ARG abundance (log10)',) +  ggtitle('Multidrug__mexB') +
  # scale_y_continuous(limits=c(-2,2))+
  theme_pubr(base_size = 23,legend ='none',x.text.angle = 30) + 
  ggeasy::easy_center_title() + 
  scale_color_npg()


### ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
my_comparisons <- list(c("Yes", "No"))

df_combined$tetracycline__tetB_log10 <- log10(df_combined$tetracycline__tetB)
ggboxplot(df_combined, x="Diagnosed.Pneumonia", y="tetracycline__tetB_log10",
          color = "Diagnosed.Pneumonia", #palette =c("#0073C2", "#EFC000","#ffb500"),
          #   add = "dotplot",add.params = list(binwidth = 0.1, dotsize = 0.03))  +
          add = "jitter",add.params = list(size = 2.4, alpha = 0.8))  +
  geom_boxplot(aes(color=Diagnosed.Pneumonia), lwd = 0.7) +
  # geom_boxplot(aes(color=Diagnosed.Pneumonia), lwd = 0.7) +
  scale_x_discrete(labels = c("No" = "No Pneumonia", "Yes" = "Pneumonia")) +
  #   stat_boxplot(mapping = aes(color=Diagnosed.Pneumonia),geom = "errorbar",width=0.05)  +
  stat_compare_means(comparisons = my_comparisons,label = "p.label",size=7,
                     vjust=.15,tip.length= c(0,0,0)) + 
  #   yscale("log10",.format = T)  +
  labs(x=NULL,y='Total ARG abundance (log10)',) +  ggtitle('Tetracycline__tetB') +
  # scale_y_continuous(limits=c(-2,2))+
  theme_pubr(base_size = 23,legend ='none',x.text.angle = 30) + 
  ggeasy::easy_center_title() + 
  scale_color_npg()



### ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
my_comparisons <- list(c("Yes", "No"))

df_combined$aminoglycoside__aph.3...IIb_log10 <- log10(df_combined$aminoglycoside__aph.3...IIb)
ggboxplot(df_combined, x="Diagnosed.Pneumonia", y="aminoglycoside__aph.3...IIb_log10",
          color = "Diagnosed.Pneumonia", #palette =c("#0073C2", "#EFC000","#ffb500"),
          #   add = "dotplot",add.params = list(binwidth = 0.1, dotsize = 0.03))  +
          add = "jitter",add.params = list(size = 2.4, alpha = 0.8))  +
  geom_boxplot(aes(color=Diagnosed.Pneumonia), lwd = 0.7) +
  # geom_boxplot(aes(color=Diagnosed.Pneumonia), lwd = 0.7) +
  scale_x_discrete(labels = c("No" = "No Pneumonia", "Yes" = "Pneumonia")) +
  #   stat_boxplot(mapping = aes(color=Diagnosed.Pneumonia),geom = "errorbar",width=0.05)  +
  stat_compare_means(comparisons = my_comparisons,label = "p.label",size=7,
                     vjust=.15,tip.length= c(0,0,0)) + 
  #   yscale("log10",.format = T)  +
  labs(x=NULL,y='Total ARG abundance (log10)',) +  ggtitle('Aminoglycoside__aph.3...IIb') +
  # scale_y_continuous(limits=c(-2,2))+
  theme_pubr(base_size = 23,legend ='none',x.text.angle = 30) + 
  ggeasy::easy_center_title() + 
  scale_color_npg()

```

```{r Fig 2i,j; Fig S3h}

df_target_argsubtype_icu_day  <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Main_figures\argsubtype-pneumonia-icu_day.csv)')
df_target_argsubtype_icu_day$multidrug__mexE_log10 <- log10(df_target_argsubtype_icu_day$multidrug__mexE)
df_target_argsubtype_icu_day$multidrug__smeB_log10 <- log10(df_target_argsubtype_icu_day$multidrug__smeB)
df_target_argsubtype_icu_day$aminoglycoside__aph.3...IIb_log10 <- log10(df_target_argsubtype_icu_day$aminoglycoside__aph.3...IIb)
df_target_argsubtype_icu_day$multidrug__smeC_log10 <- log10(df_target_argsubtype_icu_day$multidrug__smeC)


### Box plot for each argsubtype
my_comparisons <- list(c("[0,7]", ">28"), c("(7,14]", ">28"), c( "(14,21]",">28"), c("(21,28]", ">28"))
# my_comparisons <- list(c("(7,14]","[0,7]"), c("(14,21]","[0,7]"), c( "(21,28]","[0,7]"), c(">28","[0,7]"))
df_target_argsubtype_icu_day$Sample.Day_ICU_fac2 <- factor(df_target_argsubtype_icu_day$Sample.Day_ICU_fac2, levels = c("[0,7]", "(7,14]", "(14,21]", "(21,28]", ">28"))


ggboxplot(df_target_argsubtype_icu_day, x="Sample.Day_ICU_fac2", y="multidrug__mexE_log10",
          color = "Sample.Day_ICU_fac2", #palette =c("#0073C2", "#EFC000","#ffb500"),
        #   add = "dotplot",add.params = list(binwidth = 0.1, dotsize = 0.03))  +
          add = "jitter",add.params = list(size = 2.4, alpha = 0.8))  +
  geom_boxplot(aes(color=Sample.Day_ICU_fac2), lwd = 0.7) +
  # geom_boxplot(aes(color=Sample.Day_ICU_fac2), lwd = 0.7) +
  # scale_x_discrete(labels = c("No" = "No Pneumonia", "Yes" = "Pneumonia")) +
#   stat_boxplot(mapping = aes(color=Diagnosed.Pneumonia),geom = "errorbar",width=0.05)  +
  stat_compare_means(comparisons = my_comparisons,label = "p.label",size=7,
                     vjust=.15,tip.length= c(0,0,0)) + 
#   yscale("log10",.format = T)  +
  labs(x='Days in the ICU',y='Total ARG abundance (log10)',) +  ggtitle('Multidrug__mexE') +
  # scale_y_continuous(limits=c(-2,2))+
  theme_pubr(base_size = 23,legend ='none',x.text.angle = 50) + 
  ggeasy::easy_center_title() + 
  scale_color_npg()

#---------------------------------------------------------------------------------------

ggboxplot(df_target_argsubtype_icu_day, x="Sample.Day_ICU_fac2", y="multidrug__smeB_log10",
          color = "Sample.Day_ICU_fac2", #palette =c("#0073C2", "#EFC000","#ffb500"),
        #   add = "dotplot",add.params = list(binwidth = 0.1, dotsize = 0.03))  +
          add = "jitter",add.params = list(size = 2.4, alpha = 0.8))  +
  geom_boxplot(aes(color=Sample.Day_ICU_fac2), lwd = 0.7) +
  # geom_boxplot(aes(color=Sample.Day_ICU_fac2), lwd = 0.7) +
  # scale_x_discrete(labels = c("No" = "No Pneumonia", "Yes" = "Pneumonia")) +
#   stat_boxplot(mapping = aes(color=Diagnosed.Pneumonia),geom = "errorbar",width=0.05)  +
  stat_compare_means(comparisons = my_comparisons,label = "p.label",size=7,
                     vjust=.15,tip.length= c(0,0,0)) + 
#   yscale("log10",.format = T)  +
  labs(x='Days in the ICU',y='Total ARG abundance (log10)',) +  ggtitle('Multidrug__smeB') +
  # scale_y_continuous(limits=c(-2,2))+
  theme_pubr(base_size = 23,legend ='none',x.text.angle = 50) + 
  ggeasy::easy_center_title() + 
  scale_color_npg()


#---------------------------------------------------------------------------------------

ggboxplot(df_target_argsubtype_icu_day, x="Sample.Day_ICU_fac2", y="aminoglycoside__aph.3...IIb_log10",
          color = "Sample.Day_ICU_fac2", #palette =c("#0073C2", "#EFC000","#ffb500"),
        #   add = "dotplot",add.params = list(binwidth = 0.1, dotsize = 0.03))  +
          add = "jitter",add.params = list(size = 2.4, alpha = 0.8))  +
  geom_boxplot(aes(color=Sample.Day_ICU_fac2), lwd = 0.7) +
  # geom_boxplot(aes(color=Sample.Day_ICU_fac2), lwd = 0.7) +
  # scale_x_discrete(labels = c("No" = "No Pneumonia", "Yes" = "Pneumonia")) +
#   stat_boxplot(mapping = aes(color=Diagnosed.Pneumonia),geom = "errorbar",width=0.05)  +
  stat_compare_means(comparisons = my_comparisons,label = "p.label",size=7,
                     vjust=.15,tip.length= c(0,0,0)) + 
#   yscale("log10",.format = T)  +
  labs(x='Days in the ICU',y='Total ARG abundance (log10)',) +  ggtitle('Aminoglycoside__aph.3...IIb') +
  # scale_y_continuous(limits=c(-2,2))+
  theme_pubr(base_size = 23,legend ='none',x.text.angle = 50) + 
  ggeasy::easy_center_title() + 
  scale_color_npg()

#---------------------------------------------------------------------------------------

ggboxplot(df_target_argsubtype_icu_day, x="Sample.Day_ICU_fac2", y="multidrug__smeC_log10",
          color = "Sample.Day_ICU_fac2", #palette =c("#0073C2", "#EFC000","#ffb500"),
        #   add = "dotplot",add.params = list(binwidth = 0.1, dotsize = 0.03))  +
          add = "jitter",add.params = list(size = 2.4, alpha = 0.8))  +
  geom_boxplot(aes(color=Sample.Day_ICU_fac2), lwd = 0.7) +
  # geom_boxplot(aes(color=Sample.Day_ICU_fac2), lwd = 0.7) +
  # scale_x_discrete(labels = c("No" = "No Pneumonia", "Yes" = "Pneumonia")) +
#   stat_boxplot(mapping = aes(color=Diagnosed.Pneumonia),geom = "errorbar",width=0.05)  +
  stat_compare_means(comparisons = my_comparisons,label = "p.label",size=7,
                     vjust=.15,tip.length= c(0,0,0)) + 
#   yscale("log10",.format = T)  +
  labs(x='Days in the ICU',y='Total ARG abundance (log10)',) +  ggtitle('Multidrug__smeC') +
  # scale_y_continuous(limits=c(-2,2))+
  theme_pubr(base_size = 23,legend ='none',x.text.angle = 50) + 
  ggeasy::easy_center_title() + 
  scale_color_npg()


```
### Figure S2
### --------------------------------------------------------------------------------------------------------------------------------------------


```{r Fig S2a}

df_species_selected <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Main_figures\MDS_metrics-bray+species_abundance-species_pneumonia.csv)') # save in the same file 'MDS_metrics-bray-species_profiles_pneumonia.txt'
df_species_selected$SamplingSite  <- str_replace_all(df_species_selected$SamplingSite, "-", "")
df_species_selected$variable <- str_split(df_species_selected$variable, "__", simplify = TRUE)[,2]
df_species_selected$variable  <- str_replace_all(df_species_selected$variable, "_", " ")
df_species_selected$variable  <- str_replace_all(df_species_selected$variable, " group", " ")

my_comparisons <- list( c("HospitalA", "HospitalB"), c("HospitalA", "HospitalC"), c("HospitalB", "HospitalC") )
df_species_selected$SamplingSite <- factor(df_species_selected$SamplingSite,levels = c("HospitalA", "HospitalB","HospitalC"))

p_species_selected <- ggboxplot(df_species_selected, "SamplingSite", "value",
                             color = "SamplingSite", palette =c("#e64833", "#01a188", "#3e5386"),
                             add = "jitter",add.params = list(size = 2, alpha = 0.5))  +
  stat_boxplot(mapping = aes(color=SamplingSite),geom = "errorbar",width=0.05)  +
  stat_compare_means(comparisons = my_comparisons,label = "p.label",
                     size=5.0,vjust=.1,tip.length= c(0,0,0)) + ylim(-1,135) +
  facet_grid(~variable) + labs(x=NULL,y='Relative abundance (%)') +
  theme_pubr(base_size = 15,border = T,legend ='none',x.text.angle = 30) +
  theme(strip.text.x = element_text(size = 14))      
p_species_selected

```

```{r Fig S2b}

df_top_pathabundance <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\humann3-top_pathabundance-each_hospital-merged_top.csv)')
df_metadata <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\humann3_metadata.csv)')
matrix_top_abundance <- as.matrix(df_top_pathabundance[,-1])
row.names(matrix_top_abundance) <- df_top_pathabundance$pathway
matrix_top_abundance_log <- log10(matrix_top_abundance)
matrix_top_abundance_log[is.infinite(matrix_top_abundance_log)] <- 0
plot_top_pathabundance <- Heatmap(matrix_top_abundance_log,show_column_names = T,show_row_names = T,cluster_columns = F,cluster_rows = T,
        rect_gp = gpar(col= "grey"),name='Abundance (log10)',
        row_names_gp = gpar(fontsize=20),column_names_gp = gpar(fontsize=20),
        heatmap_legend_param = list(title_gp = gpar(fontsize=20),labels_gp = gpar(fontsize=20)))

```

```{r Fig S2c}

df_tempral_dyanmics_group_new_criteria <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\OI+Neutrophils-temporal_dynamics_groups.csv)')
df_tempral_dyanmics_group_new_criteria$Group <- factor(df_tempral_dyanmics_group_new_criteria$Group,levels =c('multiple','stable') )
df_tempral_dyanmics_groups_no_na <- na.omit(df_tempral_dyanmics_groups)
my_comparisons <- list(c('multiple','stable'))
df_tempral_dyanmics_group_new_criteria_OI <- df_tempral_dyanmics_group_new_criteria[!is.na(df_tempral_dyanmics_group_new_criteria$Oxygenation.Index.of.The.Day), ]

ggboxplot(df_tempral_dyanmics_group_new_criteria_OI, x="Group", y="Oxygenation.Index.of.The.Day",
          color = "Group",palette =c("#E8080A", "#0A74B2"),
          add = "jitter",add.params = list(size =2.4, alpha = 0.8))  +
  geom_boxplot(aes(color=Group), lwd = 0.7) +
  scale_x_discrete(labels = c("multiple" = "Multi-dominant", "stable" = "Mono-dominant")) +
  stat_compare_means(comparisons = my_comparisons,label = "p.label",size=7,
                     vjust=.15,tip.length= c(0,0,0)) + 
  labs(x=NULL,y='Oxygenation index',)  +
  theme_pubr(base_size = 21,legend ='none',x.text.angle = 30) 

df_tempral_dyanmics_group_new_criteria_neutrophil <- df_tempral_dyanmics_group_new_criteria[!is.na(df_tempral_dyanmics_group_new_criteria$The.Percentage.of.Neutrophils.on.The.Day.of.Sampling), ]
ggboxplot(df_tempral_dyanmics_group_new_criteria_neutrophil, x="Group", y="The.Percentage.of.Neutrophils.on.The.Day.of.Sampling",
          color = "Group",palette =c("#E8080A", "#0A74B2"), 
          add = "jitter",add.params = list(size =2.4, alpha = 0.8))  +
  geom_boxplot(aes(color=Group), lwd = 0.7) +
  scale_x_discrete(labels = c("multiple" = "Multi-dominant", "stable" = "Mono-dominant")) +
  stat_compare_means(comparisons = my_comparisons,label = "p.label",size=7,
                     vjust=.15,tip.length= c(0,0,0)) + 
  labs(x=NULL,y='Percentage of neutrophils (%)',) +  
  theme_pubr(base_size = 21,legend ='none',x.text.angle = 30)

```

### Figure 3
### --------------------------------------------------------------------------------------------------------------------------------------------

```{r Fig 3a}

df_arg_num = read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Main_figures\RGI_profile-comparisons_arg_number.csv)") # save in the same file 'RGI+VF_profile-comparisons_number.txt'
df_arg_num$species <- factor(df_arg_num$species,levels = c('pseudomonas','acinetobacter','klebsiella','corynebacterium','stenotrophomonas'))
my_comparisons <- list(c('pseudomonas','stenotrophomonas'),c('pseudomonas','corynebacterium'),
                       c('acinetobacter','stenotrophomonas'),c('acinetobacter','corynebacterium'),
                       c('klebsiella','stenotrophomonas'),c('klebsiella','corynebacterium'))

map <-c('acinetobacter'=3.965,'pseudomonas'=6.644,'klebsiella'=5.596,'corynebacterium'=2.841,'stenotrophomonas'=4.613)
df_arg_num <- df_arg_num %>%
  mutate(genome_length = map[species])
df_arg_num <- df_arg_num %>%
  mutate(arg_num_normalized = arg_num/genome_length)

ggboxplot(df_arg_num, x="species", y="arg_num_normalized",
          color = "species", palette =c("#E41A1C","#377EB8","#4DAF4A","#984EA3","#FF7F00","#F781BF","#A65628","#F781BF"),
          add = "jitter",add.params = list(size = 2, alpha = 0.5))  +
  stat_boxplot(mapping = aes(color=species),geom = "errorbar",width=0.1,lwd=0.8) +
  stat_compare_means(comparisons = rev(my_comparisons),label = "p.label",size=10,
                     vjust=.15,tip.length= c(0,0,0)) +
  scale_x_discrete(labels=c('P. aeruginosa','A. baumannii','K. pneumoniae','C. striatum','S. maltophilia'))+
  labs(x=NULL,y='Number of ARGs (Normalized)',) +  
  theme_pubr(base_size = 32,legend ='none',x.text.angle = 30) 


```

```{r Fig 3b}


df_rgi_species_all <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Main_figures\RGI_profile_species_all.csv)') # save in the same file 'RGI_profile_species_all.txt'
row.names(df_rgi_species_all) <- df_rgi_species_all$filename
matrix_rgi_species_all <- as.matrix(df_rgi_species_all[,-1])
df_metadata_species_all <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Main_figures\RGI_profile_species_all_metadata.csv)') # save in the same file 'RGI_profile_species_all.txt'
labels_annotation_col <- data.frame(species=df_metadata_species_all$species)
# colnames(labels_annotation_col) <- c('species')
row.names(labels_annotation_col) <- df_metadata_species_all$filename

labels_annotation_samplingsite <- data.frame(species=df_metadata_species_all$SamplingSite)
row.names(labels_annotation_samplingsite) <- df_metadata_species_all$filename

tree_species_all <- read.tree(r"(LRT_microbiomes_critical_illness\Data_repository\Main_figures\RGI_profile_species_all.tre.nwk)") # save in the same file 'RGI_profile_species_all.txt'
order_row <- data.frame(species=tree_species_all$tip.label)
order_row <- na.omit(order_row)
#horizontal version
plot_horizontal <- Heatmap(matrix_rgi_species_all,
        #breaks = c(0,1,3),
        col = c('#ffffff','#ff0000'),
        left_annotation   = rowAnnotation(df=labels_annotation_col,
                                         col=list(species=c(corynebacterium="#e36255", stenotrophomonas="#ec9a86",
                                                            pseudomonas="#a2c5c9", klebsiella="#f3c262", 
                                                            acinetobacter="#f1e0ce")),
                                         show_annotation_name = F,show_legend=F),
        right_annotation = rowAnnotation(df=labels_annotation_samplingsite,
                                         col=list(species=c(HospitalA="#32A251", HospitalB="#008FD5",
                                                            HospitalC="#FC7D0B")),
                                         show_annotation_name = F,show_legend=F),
        cluster_columns =T,
        cluster_rows =F,
        show_column_dend = F,show_row_dend = F,#rect_gp = gpar(col= "#B9B9B9",lwd=0.02),
        show_column_names =F,show_row_names =F,
        show_heatmap_legend = F,
        column_names_gp = grid::gpar(fontsize=3),
        row_order = order_row$species )

```

```{r Fig 3c}

###plot RGI profile for acinetobacter
df_rgi_profile_acinetobacter = read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Main_figures\RGI_profile-PCoA_acinetobacter.csv)", header= TRUE) # save in the same file 'RGI_profile-PCoA_combined.txt'
abundance_acinetobacter = df_rgi_profile_acinetobacter[,2:(ncol(df_rgi_profile_acinetobacter)-2)]
m_abundance_acinetobacter = as.matrix(abundance_acinetobacter)


diss_acinetobacter <- vegdist(m_abundance_acinetobacter, method = "bray")
pcoa_acinetobacter <- cmdscale(diss_acinetobacter, eig = TRUE, k = 2) 

data_scores_acinetobacter = as.data.frame(pcoa_acinetobacter$points)
names(data_scores_acinetobacter) <- c("PCoA1", "PCoA2")
k_acinetobacter <- 4 
kmeans_result_acinetobacter <- kmeans(as.data.frame(pcoa_acinetobacter$points), centers = k_acinetobacter)

data_scores_acinetobacter$cluster = kmeans_result_acinetobacter$cluster
data_scores_acinetobacter$cluster <- factor(data_scores_acinetobacter$cluster)

p_scatter_acinetobacter <- ggscatter(data_scores_acinetobacter,
                                     x = "PCoA1", 
                                     y = "PCoA2",
                                     color = 'cluster',
                                     palette = c("#7f59af", "#e64833", "#63c5ea", "#3e5386"),
                                     #shape = "SamplingSite",
                                     size=5,
                                     ggtheme = theme_pubr(border = T,base_size = 45),
                                     legend = 'none',
                                     ellipse = T,ellipse.level = 0.90)  + ggtitle('A. baumannii') +
  ggeasy::easy_center_title() + theme(plot.title = element_text(face = "italic",size = 45))

###-----------------------------------------------------------------------------------------------------------------------------------------

###plot RGI profile for corynebacterium
df_rgi_profile_corynebacterium = read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Main_figures\RGI_profile-PCoA_corynebacterium.csv)", header= TRUE) # save in the same file 'RGI_profile-PCoA_combined.txt'
abundance_corynebacterium = df_rgi_profile_corynebacterium[,2:(ncol(df_rgi_profile_corynebacterium)-2)]
m_abundance_corynebacterium = as.matrix(abundance_corynebacterium)

diss_corynebacterium <- vegdist(m_abundance_corynebacterium, method = "bray")
pcoa_corynebacterium <- cmdscale(diss_corynebacterium, eig = TRUE, k = 2) 

data_scores_corynebacterium = as.data.frame(pcoa_corynebacterium$points)
names(data_scores_corynebacterium) <- c("PCoA1", "PCoA2")
k_corynebacterium <- 3  
kmeans_result_corynebacterium <- kmeans(as.data.frame(pcoa_corynebacterium$points), centers = k_corynebacterium)

data_scores_corynebacterium$cluster = kmeans_result_corynebacterium$cluster
data_scores_corynebacterium$cluster <- factor(data_scores_corynebacterium$cluster)

p_scatter_corynebacterium <- ggscatter(data_scores_corynebacterium,
                                       x = "PCoA1", 
                                       y = "PCoA2",
                                       color = 'cluster',
                                       palette = c("#7f59af", "#e64833", "#63c5ea", "#3e5386"),
                                       #shape = "SamplingSite",
                                       size=5,
                                       ggtheme = theme_pubr(border = T,base_size = 45),
                                       legend = 'none',
                                       ellipse = T,ellipse.level = 0.90)  + ggtitle('C. striatum') +
  ggeasy::easy_center_title() + theme(plot.title = element_text(face = "italic",size = 45))

###-----------------------------------------------------------------------------------------------------------------------------------------

###plot RGI profile for klebsiella
df_rgi_profile_klebsiella = read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Main_figures\RGI_profile-PCoA_klebsiella.csv)", header= TRUE) # save in the same file 'RGI_profile-PCoA_combined.txt'
abundance_klebsiella = df_rgi_profile_klebsiella[,2:(ncol(df_rgi_profile_klebsiella)-2)]
m_abundance_klebsiella = as.matrix(abundance_klebsiella)

diss_klebsiella <- vegdist(m_abundance_klebsiella, method = "bray")
pcoa_klebsiella <- cmdscale(diss_klebsiella, eig = TRUE, k = 2) 

data_scores_klebsiella = as.data.frame(pcoa_klebsiella$points)
names(data_scores_klebsiella) <- c("PCoA1", "PCoA2")
kmeans_result_klebsiella <- kmeans(as.data.frame(pcoa_klebsiella$points), centers = k_klebsiella)
data_scores_klebsiella$cluster = kmeans_result_klebsiella$cluster
data_scores_klebsiella$cluster <- factor(data_scores_klebsiella$cluster)

p_scatter_klebsiella <- ggscatter(data_scores_klebsiella,
                                  x = "PCoA1", 
                                  y = "PCoA2",
                                  color = 'cluster',
                                  palette = c("#7f59af", "#e64833", "#63c5ea", "#3e5386"),
                                  #shape = "SamplingSite",
                                  size=5,
                                  ggtheme = theme_pubr(border = T,base_size = 45),
                                  legend = 'none',
                                  ellipse = T,ellipse.level = 0.90)  + ggtitle('K. pneumoniae') +
  ggeasy::easy_center_title() + theme(plot.title = element_text(face = "italic",size = 45))

(p_scatter_acinetobacter)/(p_scatter_corynebacterium)/(p_scatter_klebsiella)


```

```{r Fig 3d}


df_vf_num = read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Main_figures\vfdb_profile-species_all-vf_num.csv)")  # save in the same file 'RGI+VF_profile-comparisons_number.txt'
df_vf_num$species <- factor(df_vf_num$species,levels = c('pseudomonas','acinetobacter','klebsiella','corynebacterium','stenotrophomonas'))
my_comparisons <- list(c('pseudomonas','stenotrophomonas'),c('pseudomonas','corynebacterium'),
                       c('acinetobacter','stenotrophomonas'),c('acinetobacter','corynebacterium'),
                       c('klebsiella','stenotrophomonas'),c('klebsiella','corynebacterium'))

map <-c('acinetobacter'=3.965,'pseudomonas'=6.644,'klebsiella'=5.596,'corynebacterium'=2.841,'stenotrophomonas'=4.613)
df_vf_num <- df_vf_num %>%
  mutate(genome_length = map[species])
df_vf_num <- df_vf_num %>%
  mutate(vf_num_normalized = vf_num/genome_length)

ggboxplot(df_vf_num, x="species", y="vf_num_normalized",
          color = "species", palette =c("#E41A1C","#377EB8","#4DAF4A","#984EA3","#FF7F00","#F781BF","#A65628","#F781BF"),
          add = "jitter",add.params = list(size = 1.5, alpha = 0.5))  +
  stat_boxplot(mapping = aes(color=species),geom = "errorbar",width=0.1,lwd=0.8) +
  stat_compare_means(comparisons = rev(my_comparisons),label = "p.label",size=10,
                     vjust=.15,tip.length= c(0,0,0)) +
  scale_x_discrete(labels=c('P. aeruginosa','A. baumannii','K. pneumoniae','C. striatum','S. maltophilia'))+
  labs(x=NULL,y='Number of VFs (Normalized)',) +  
  theme_pubr(base_size = 32,legend ='none',x.text.angle = 30) 

```

```{r Fig 3e}

df_vf_species_all <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Main_figures\vfdb_profile-species_all-VFG.csv)')  # save in the same file 'vf_profile_species_all.txt'
row.names(df_vf_species_all) <- df_vf_species_all$X.FILE
df_vf_species_all[df_vf_species_all>0] =1
matrix_vf_species_all <- as.matrix(df_vf_species_all[,-c(1,2)])
df_metadata_species_all <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Main_figures\RGI_profile_species_all_metadata.csv)')  # save in the same file 'vf_profile_species_all.txt'
labels_annotation_col <- data.frame(species=df_metadata_species_all$species)
row.names(labels_annotation_col) <- df_metadata_species_all$filename

labels_annotation_samplingsite <- data.frame(species=df_metadata_species_all$SamplingSite)
row.names(labels_annotation_samplingsite) <- df_metadata_species_all$filename

tree_species_all <- read.tree(r"(LRT_microbiomes_critical_illness\Data_repository\Main_figures\vfdb_profile-species_all.tre.nwk)")  # save in the same file 'vf_profile_species_all.txtt'
order_row <- data.frame(species=tree_species_all$tip.label)

plot_heatmap_species_all <- Heatmap(matrix_vf_species_all,
                                    #breaks = c(0,1,3),
                                    col = c('#ffffff','#ff0000'),
                                    # col = c('#f7faff','#08306b'),
                                    left_annotation  = rowAnnotation(df=labels_annotation_col,
                                                                     col=list(species=c(corynebacterium="#e36255", stenotrophomonas="#ec9a86",
                                                                                        pseudomonas="#a2c5c9", klebsiella="#f3c262",
                                                                                        acinetobacter="#f1e0ce")),show_legend=F),
                                    right_annotation = rowAnnotation(df=labels_annotation_samplingsite,
                                                                         col=list(species=c(HospitalA="#32A251", HospitalB="#008FD5",
                                                                                            HospitalC="#FC7D0B")),
                                                                         show_annotation_name = T,show_legend=F),
                                    cluster_rows =T,cluster_columns =F,
                                    show_column_dend = F,show_row_dend =F,#rect_gp = gpar(col= "#B9B9B9",lwd=0.000001),
                                    show_column_names = F,show_row_names = F,
                                    show_heatmap_legend = F,
                                    row_order = order_row$species)

```

```{r Fig 3f}

###plot vfdb profile for acinetobacter
df_vfdb_profile_acinetobacter = read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Main_figures\vfdb_profile-PCoA_acinetobacter.csv)", header= TRUE) # save in the same file 'vf_profile-PCoA_combined.txt'
abundance_acinetobacter = df_vfdb_profile_acinetobacter[,2:(ncol(df_vfdb_profile_acinetobacter)-2)]
m_abundance_acinetobacter = as.matrix(abundance_acinetobacter)


diss_acinetobacter <- vegdist(m_abundance_acinetobacter, method = "bray")
pcoa_acinetobacter <- cmdscale(diss_acinetobacter, eig = TRUE, k = 2) 
data_scores_acinetobacter = as.data.frame(pcoa_acinetobacter$points)
names(data_scores_acinetobacter) <- c("PCoA1", "PCoA2")
k_acinetobacter <- 5  
kmeans_result_acinetobacter <- kmeans(as.data.frame(pcoa_acinetobacter$points), centers = k_acinetobacter)
data_scores_acinetobacter$cluster = kmeans_result_acinetobacter$cluster
data_scores_acinetobacter$cluster <- factor(data_scores_acinetobacter$cluster)

p_scatter_acinetobacter <- ggscatter(data_scores_acinetobacter,
                                     x = "PCoA1", 
                                     y = "PCoA2",
                                     color = 'cluster',
                                     palette = c("#7f59af", "#e64833", "#63c5ea", "#3e5386",'#f6cbcb'),
                                     #shape = "SamplingSite",
                                     size=5,
                                     ggtheme = theme_pubr(border = T,base_size = 45),
                                     legend = 'none',
                                     ellipse = T,ellipse.level = 0.90)  + ggtitle('A. baumannii') +
  ggeasy::easy_center_title() + theme(plot.title = element_text(face = "italic",size = 45))



###-----------------------------------------------------------------------------------------------------------------------------------------

###plot vfdb profile for klebsiella
df_vfdb_profile_klebsiella = read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Main_figures\vfdb_profile-PCoA_klebsiella.csv)", header= TRUE) # save in the same file 'vf_profile-PCoA_combined.txt'
abundance_klebsiella = df_vfdb_profile_klebsiella[,2:(ncol(df_vfdb_profile_klebsiella)-2)]
m_abundance_klebsiella = as.matrix(abundance_klebsiella)
diss_klebsiella <- vegdist(m_abundance_klebsiella, method = "bray")
pcoa_klebsiella <- cmdscale(diss_klebsiella, eig = TRUE, k = 2) 
data_scores_klebsiella = as.data.frame(pcoa_klebsiella$points)
names(data_scores_klebsiella) <- c("PCoA1", "PCoA2")
k_klebsiella <- 3  
kmeans_result_klebsiella <- kmeans(as.data.frame(pcoa_klebsiella$points), centers = k_klebsiella)
data_scores_klebsiella$cluster = kmeans_result_klebsiella$cluster
data_scores_klebsiella$cluster <- factor(data_scores_klebsiella$cluster)

p_scatter_klebsiella <- ggscatter(data_scores_klebsiella,
                                  x = "PCoA1", 
                                  y = "PCoA2",
                                  color = 'cluster',
                                  palette = c("#7f59af", "#e64833", "#63c5ea", "#3e5386"),
                                  #shape = "SamplingSite",
                                  size=5,
                                  ggtheme = theme_pubr(border = T,base_size = 45),
                                  legend = 'none',
                                  ellipse = T,ellipse.level = 0.90)  + ggtitle('K. pneumoniae') +
  ggeasy::easy_center_title() + theme(plot.title = element_text(face = "italic",size = 45))


###-----------------------------------------------------------------------------------------------------------------------------------------

###plot vfdb profile for pseudomonas
df_vfdb_profile_pseudomonas = read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Main_figures\vfdb_profile-PCoA_pseudomonas.csv)", header= TRUE) # save in the same file 'vf_profile-PCoA_combined.txt'
abundance_pseudomonas = df_vfdb_profile_pseudomonas[,2:(ncol(df_vfdb_profile_pseudomonas)-2)]
m_abundance_pseudomonas = as.matrix(abundance_pseudomonas)
diss_pseudomonas <- vegdist(m_abundance_pseudomonas, method = "bray")
pcoa_pseudomonas <- cmdscale(diss_pseudomonas, eig = TRUE, k = 2) 
data_scores_pseudomonas = as.data.frame(pcoa_pseudomonas$points)
names(data_scores_pseudomonas) <- c("PCoA1", "PCoA2")
k_pseudomonas <- 3 
kmeans_result_pseudomonas <- kmeans(as.data.frame(pcoa_pseudomonas$points), centers = k_pseudomonas)
data_scores_pseudomonas$cluster = kmeans_result_pseudomonas$cluster
data_scores_pseudomonas$cluster <- factor(data_scores_pseudomonas$cluster)

p_scatter_pseudomonas <- ggscatter(data_scores_pseudomonas,
                                   x = "PCoA1", 
                                   y = "PCoA2",
                                   color = 'cluster',
                                   palette = c("#7f59af", "#e64833", "#63c5ea", "#3e5386"),
                                   #shape = "SamplingSite",
                                   size=5,
                                   ggtheme = theme_pubr(border = T,base_size = 45),
                                   legend = 'none',
                                   ellipse = T,ellipse.level = 0.90)  + ggtitle('P. aeruginosa') +
  ggeasy::easy_center_title() + theme(plot.title = element_text(face = "italic",size = 45))

(p_scatter_acinetobacter)/(p_scatter_klebsiella)/(p_scatter_pseudomonas)


```

### Figure S3
### --------------------------------------------------------------------------------------------------------------------------------------------

```{r Fig S3a}

df_argsubtype_top3 <- read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\statistics-pneumonia_patients-argsubtypes_top-each_hospital.csv)")
df_argsubtype_top3$arg_type <- factor(df_argsubtype_top3$arg_type,levels = c('Multidrug','Beta-lactam','Aminoglycoside','MLS'))

pal <- c('Multidrug__acra'='#3182BD','Multidrug__multidrug_transporter'='#00A2CE','Beta-lactam__shv-39'='#31A354','Beta-lactam__cfxa2'='#008179',
         'Beta-lactam__oxa-225'='#9ED1A6','Aminoglycoside__rmtb'='#FFB500',"Aminoglycoside__aac(6')-i"='#FFB686',"Aminoglycoside__aph(3'')-i"='#FFEDCB',
         'MLS__maca'='#EFD7EB','MLS__ermx'='#FFD1E0')
ggbarplot(df_argsubtype_top3,
          x='sites',y='mean_abundance',fill = 'argsubtypes') + 
  scale_fill_manual(values = pal)  +
  labs(x=NULL,y='Mean abundance') +
  guides(fill=guide_legend('ARG subtype',ncol = 2)) +
  facet_grid(~arg_type) + theme_pubr(border = T,x.text.angle = 30,legend = 'bottom',base_size = 20)

```

```{r Fig S3b}

df_argtypes_collection <- read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Main_figures\argtypes-collection_all_sites-pneumonia_patients.csv)")
df_argtypes_collection2 <- arrange(df_argtypes_collection,SamplingSite)
my_comparisons <- list( c("HospitalA", "HospitalB"), c("HospitalA", "HospitalC"), c("HospitalB", "HospitalC") )


#plot for 'MLS'
ggboxplot(df_argtypes_collection2, x="SamplingSite", y="macrolide.lincosamide.streptogramin",
          color = "SamplingSite", palette =c("#f93800", "#007a8b","#ffb500"),
          add = "jitter",add.params = list(size = 1.5, alpha = 0.5))  +
  stat_boxplot(mapping = aes(color=SamplingSite),geom = "errorbar",width=0.1)  +
  stat_compare_means(comparisons = my_comparisons,label = "p.label",size=5,
                     vjust=.15,tip.length= c(0,0,0)) + 
  yscale("log10", .format = F)  + 
  labs(x=NULL,y='ARG abundance',) +  ggtitle('MLS') +
  theme_pubr(base_size = 18,legend ='none',x.text.angle = 30) +
  ggeasy::easy_center_title() 

#plot for 'tetracycline'
ggboxplot(df_argtypes_collection2, x="SamplingSite", y="tetracycline",
          color = "SamplingSite", palette =c("#f93800", "#007a8b","#ffb500"),
          add = "jitter",add.params = list(size = 1.5, alpha = 0.5))  +
  stat_boxplot(mapping = aes(color=SamplingSite),geom = "errorbar",width=0.1)  +
  stat_compare_means(comparisons = my_comparisons,label = "p.label",size=5,
                     vjust=.15,tip.length= c(0,0,0)) + 
  yscale("log10", .format = F)  + 
  labs(x=NULL,y='ARG abundance',) +  ggtitle('Tetracycline') +
  theme_pubr(base_size = 18,legend ='none',x.text.angle = 30) +
  ggeasy::easy_center_title() 

#plot for 'sulfonamide'
ggboxplot(df_argtypes_collection2, x="SamplingSite", y="sulfonamide",
          color = "SamplingSite", palette =c("#f93800", "#007a8b","#ffb500"),
          add = "jitter",add.params = list(size = 1.5, alpha = 0.5))  +
  stat_boxplot(mapping = aes(color=SamplingSite),geom = "errorbar",width=0.1)  +
  stat_compare_means(comparisons = my_comparisons,label = "p.label",size=5,
                     vjust=.15,tip.length= c(0,0,0)) + 
  yscale("log10", .format = F)  + 
  labs(x=NULL,y='ARG abundance',) +  ggtitle('Sulfonamide') +
  theme_pubr(base_size = 18,legend ='none',x.text.angle = 30) +
  ggeasy::easy_center_title() 

```

```{r Fig S3c}

df_nmds = read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\NMDS_arg_subtypes-all_sites.csv)',header = TRUE)
my_comparisons <- list( c("HospitalA", "HospitalB"), c("HospitalA", "HospitalC"), c("HospitalB", "HospitalC") )
df_nmds$NMDS1 <- as.numeric(df_nmds$NMDS1)
df_nmds$NMDS2 <- as.numeric(df_nmds$NMDS2)


#scatter plot to show the argsubtypes distribution for all ETA samples
p_scatter <- ggscatter(df_nmds,
                       x = "NMDS1", 
                       y = "NMDS2",
                       color = 'SamplingSite',
                       palette = c("#e64833", "#01a188", "#3e5386"),
                       #shape = "SamplingSite",
                       size=5,
                       ggtheme = theme_pubr(border = T,base_size = 32),
                       legend = 'none')  + xlim2(limits = c(-2.5,2.5)) + ylim2(limits = c(-2.5,1.8)) +  
  theme(axis.title = element_blank(),
        axis.ticks = element_blank(),
        axis.text.x = element_blank(),
        axis.text.y = element_blank())
p <-ggscatterhist(df_nmds,
              x = "NMDS1", 
              y = "NMDS2",
              xlim= c(-2.2,2.2),
              ylim= c(-2.5,1.5),
              color = "SamplingSite",
              size=5,
              palette = c("#e64833", "#01a188", "#3e5386"),
              margin.plot = "density",
              ggtheme = theme_pubr(border = T,base_size = 32),
              margin.params = list(fill="SamplingSite",size=0.3),
              legend = 'right')

```

```{r Fig S3d}

df_splsda = read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\PLS-DA_arg_subtypes-all_sites.csv)',header = TRUE)

### Plot the PLS-DA components distribution 
plot_splsda <- ggscatter(df_splsda,
          x = "comp1", 
          y = "comp2",
          color = "SamplingSite",
          # shape = "SamplingSite",
          size=4,
          ellipse=TRUE,
          # star.plot = TRUE,
          # palette = c("#00AFBB", "#E7B800", "#FC4E07"), #palette1
          palette = c("#c23726", "#00C19B", "#2c83c6"),  #palette2
          # palette = c("#e64833", "#01a188", "#3e5386"), #palette3
          # margin.plot = "boxplot",
          # ggtheme = theme_pubr(),
          # margin.params = list(fill="SamplingSite",size=0.36),
          legend = 'right')
ggpar(plot_splsda,
      xlab = "Component 1 (12%)",
      ylab = "Component 2 (3%)",
      font.xtickslab=c(32),
      font.ytickslab=c(32),
      font.x = c(32),
      font.y = c(32),
      legend.title = "Sampling site",
      font.legend = c(32))  + theme_pubr(base_size = 32,legend = "none",margin = FALSE,border = T)

```

```{r Fig S3e}

df_loading1 <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\PLS-DA_arg_subtypes-all_sites-loadings-comp1.csv)',header = TRUE) # save in the same file 'PLS-DA_arg_subtypes-all_sites-loadings.txt'
plot_loading1 <- ggbarplot(df_loading1[1:6,],x = "X", y = "value.var",orientation = "horiz",
          fill = "#e6a024", color = "#e6a024",sort.val = 'desc',ggtheme = theme_bw(base_size = 19)) + labs(title = 'Component-1') +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text = element_blank(),
        axis.title = element_blank())

plot_loading1 <- ggpar(plot_loading1,
      orientation = 'reverse')  #+ theme_pubr(base_size = 32,legend = "right",margin = FALSE)

#plot loadings on component-2
df_loading2 <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\PLS-DA_arg_subtypes-all_sites-loadings-comp2.csv)',header = TRUE) # save in the same file 'PLS-DA_arg_subtypes-all_sites-loadings.txt'
df_loading2 <- df_loading2 %>%
  mutate(X=str_replace(X,'_chloramphenicol acetyltransferase',''),
         X=str_replace(X,'_\\(ramA_or_sud2\\)',''))

plot_loading2 <- ggbarplot(df_loading2[1:6,],x = "X", y = "value.var",orientation = "horiz",
                           fill = "#5bb4e5", color = "#5bb4e5",sort.val = 'asc',ggtheme = theme_bw(base_size = 19))+ labs(title = 'Component-2') +
  theme(plot.title = element_text(hjust = 0.5),
        axis.text = element_blank(),
        axis.title = element_blank())

plot_loading1 | plot_loading2

```

```{r Fig S3f}

lmeRes <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\lme-argsubtypes.csv)')

# Subset and reformat the dataset
lmeRes <- subset(lmeRes, predictor %in% c("Sampling_Day_ICU", "Oxygenation.Index.of.The.Day", "Diagnosed.PneumoniaYes"))
lmeRes <- lmeRes %>%
  mutate(predictor_reformat = recode(predictor, 
                            "Sampling_Day_ICU" = "Days in the ICU", 
                            "Oxygenation.Index.of.The.Day" = "Oxygenation index", 
                            "Diagnosed.PneumoniaYes" = "Diagnosed pneumonia"))

# Setting the factor levels
lmeRes$predictor_reformat <- factor(lmeRes$predictor_reformat, levels = c("Diagnosed pneumonia", "Days in the ICU", "Oxygenation index"))

valid_x <- c("Days in the ICU", "Oxygenation index", "Diagnosed pneumonia")

multiVarSigDF <- subset(lmeRes, predictor_reformat %in% valid_x & p_adjusted <= 0.05)
multiVarSigDF$p_adjusted_log <- -log10(multiVarSigDF$p_adjusted)

uniqueVar <- multiVarSigDF[, c('item', 'predictor_reformat', 'p_adjusted_log', 'p_adjusted')]
spW <- uniqueVar %>% group_by(item) %>% summarise(value = sum(p_adjusted_log))
xW <- uniqueVar %>% group_by(predictor_reformat) %>% summarise(value = sum(p_adjusted_log))

YsName <- spW$item[order(spW$value)]
# XsName <- xW$predictor_reformat[order(match(xW$predictor_reformat, valid_x))]
XsName <- rev(c("Diagnosed pneumonia", "Days in the ICU", "Oxygenation index"))

ord <- c(XsName, YsName)

grid.col <- c(rep('#FDCA4E' %>% alpha(., 1), length(YsName)),
              rep('#89ADD4' %>% alpha(., 1), length(XsName)))
names(grid.col) <- c(YsName, XsName)

color_lookup <- c(
  "Diagnosed pneumonia" = alpha("#EB9E9A", 0.7),
  "Days in the ICU" = alpha("#BAA8CE", 0.7),
  "Oxygenation index" = alpha("#DFE7F1", 0.7)
)

col_fun <- data.frame(
  uniqueVar$item,
  uniqueVar$predictor_reformat
)
col_fun$col <- color_lookup[uniqueVar$predictor_reformat]
col_fun$col[is.na(col_fun$col)] <- '#5D8DC3' %>% alpha(., 0.5)  # Default color for predictors not in the lookup

chordDiagram(uniqueVar[, c('predictor_reformat', 'item', 'p_adjusted_log')], 
             order = ord, 
             annotationTrack = "grid", 
             link.sort = TRUE,
             grid.col = grid.col, 
             col = col_fun$col, 
             transparency = 0,
             annotationTrackHeight = mm_h(2),
             preAllocateTracks = list(track.height = 0.3))

circos.track(track.index = 1,
             panel.fun = function(x, y) {
               circos.text(CELL_META$xcenter, CELL_META$ylim[2] - 0.92,  # Adjust the + 0.15 to move the text more outwards if required
                           CELL_META$sector.index, 
                           facing = "clockwise", 
                           niceFacing = TRUE, 
                           adj = c(0, 0.5), 
                           cex = 1.0)
             }, bg.border = NA)

```

```{r Fig S3g}
### code was provided in Fig 2g,h
```

```{r Fig S3h}
### code was provided in Fig 2i,j
```
### Figure 4
### --------------------------------------------------------------------------------------------------------------------------------------------

```{r Fig 4a}

df_statistics <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Main_figures\mobile_elements_finder-category_statistics+sem.csv)')
df_statistics['MGE_per_genome_round'] <- round(df_statistics['MGE_per_genome'],2)

limits <- aes(ymax = MGE_per_genome+standard_error,
              ymin = ifelse(MGE_per_genome-standard_error<0,0,MGE_per_genome-standard_error))


df_statistics_acinetobacter <- subset(df_statistics,species=='acinetobacter')
p_acinetobacter <- ggplot(df_statistics_acinetobacter, aes(x=mge_type, y=MGE_per_genome)) + 
  geom_errorbar(limits, width=.2,col='#d08b70',
                position=position_dodge(.9)) +
  geom_bar(stat = "identity",fill='#d08b70') + 
  # geom_text(aes(label=MGE_per_genome_round),position=position_dodge(width=0.9), vjust=-0.25) + 
  scale_x_discrete(limits=c('insertion sequence','unit transposon','composite transposon','mite','ice')) +
  ylim2(limits = c(0,10)) + xlab(NULL) + ylab(NULL) + labs(title = 'A. baumannii') +
  theme_bw()  +
  theme(axis.text.x  = element_blank(),axis.text.y = element_text(size=36),axis.ticks.x = element_blank()) + 
  theme(strip.background = element_blank(),
        plot.title  = element_textbox(
          size = 34,
          color = "#4d4d4d", fill = "#dbdad9", box.color = "black",face = 'italic',
          halign = 0.5, linetype = 1,  width = unit(1, "npc"),
          margin = margin(0, 0, 0, 0),padding = margin(6, 6, 6, 6)),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.x = element_blank()) 



df_statistics_corynebacterium <- subset(df_statistics,species=='corynebacterium')
p_corynebacterium <- ggplot(df_statistics_corynebacterium, aes(x=mge_type, y=MGE_per_genome)) + 
  geom_errorbar(limits, width=.2,col='#988b74',
                position=position_dodge(.9)) +
  geom_bar(stat = "identity",fill='#988b74') + 
  # geom_text(aes(label=MGE_per_genome_round),position=position_dodge(width=0.9), vjust=-0.25) + 
  scale_x_discrete(limits=c('insertion sequence','unit transposon','composite transposon','mite','ice')) +
  ylim2(limits = c(0,10)) + xlab(NULL) + ylab(NULL) + labs(title = 'C. striatum') +
  theme_bw()  +
  theme(axis.text.x  = element_blank(),axis.ticks.x = element_blank(),axis.ticks.y =element_blank(), axis.text.y =element_blank()) +
  theme(strip.background = element_blank(),
        plot.title  = element_textbox(
          size = 34,
          color = "#4d4d4d", fill = "#dbdad9", box.color = "black",face = 'italic',
          halign = 0.5, linetype = 1,  width = unit(1, "npc"),
          margin = margin(0, 0, 0, 0),padding = margin(6, 6, 6, 6)),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.x = element_blank()) 

df_statistics_klebsiella <- subset(df_statistics,species=='klebsiella')
p_klebsiella <- ggplot(df_statistics_klebsiella, aes(x=mge_type, y=MGE_per_genome)) + 
  geom_errorbar(limits, width=.2,col='#6fa9c8',
                position=position_dodge(.9)) +
  geom_bar(stat = "identity",fill='#6fa9c8') + 
  # geom_text(aes(label=MGE_per_genome_round),position=position_dodge(width=0.9), vjust=-0.25) + 
  scale_x_discrete(limits=c('insertion sequence','unit transposon','composite transposon','mite','ice')) +
  ylim2(limits = c(0,10)) + xlab(NULL) + ylab(NULL) + labs(title = 'K. pneumoniae') +
  theme_bw()  +
  theme(axis.text.x  = element_blank(),axis.ticks.x = element_blank(),axis.ticks.y =element_blank(), axis.text.y =element_blank()) +
  theme(strip.background = element_blank(),
        plot.title  = element_textbox(
          size = 34,
          color = "#4d4d4d", fill = "#dbdad9", box.color = "black",face = 'italic',
          halign = 0.5, linetype = 1,  width = unit(1, "npc"),
          margin = margin(0, 0, 0, 0),padding = margin(6, 6, 6, 6)),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.x = element_blank()) 

df_statistics_pseudomonas <- subset(df_statistics,species=='pseudomonas')
p_pseudomonas <- ggplot(df_statistics_pseudomonas, aes(x=mge_type, y=MGE_per_genome)) + 
  geom_errorbar(limits, width=.2,col='#98c5b4',
                position=position_dodge(.9)) +
  geom_bar(stat = "identity",fill='#98c5b4') + 
  # geom_text(aes(label=MGE_per_genome_round),position=position_dodge(width=0.9), vjust=-0.25) + 
  scale_x_discrete(limits=c('insertion sequence','unit transposon','composite transposon','mite','ice'),
                   labels=c('IS','Unit Transposon','Composite Transposon','MITE', 'ICE')) +
  ylim2(limits = c(0,10)) + xlab(NULL) + ylab('Number of mobile elements\nper genome') + labs(title = 'P. aeruginosa') +
  theme_bw()  + 
  theme(strip.background = element_blank(),
        plot.title  = element_textbox(
          size = 34,
          color = "#4d4d4d", fill = "#dbdad9", box.color = "black",face = 'italic',
          halign = 0.5, linetype = 1,  width = unit(1, "npc"),
          margin = margin(0, 0, 0, 0),padding = margin(6, 6, 6, 6)),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.x = element_blank()) + 
  theme(axis.text.x = element_text(angle = 30,hjust = 1,size=30),
        axis.text.y = element_text(size = 36),
        axis.title.y = element_text(size=36)) #change ylabs fontsize,ylab() --> axis.title.y()

df_statistics_stenotrophomonas <- subset(df_statistics,species=='stenotrophomonas')
p_stenotrophomonas <- ggplot(df_statistics_stenotrophomonas, aes(x=mge_type, y=MGE_per_genome)) + 
  geom_errorbar(limits, width=.2,col='#384073',
                position=position_dodge(.9)) +
  geom_bar(stat = "identity",fill='#384073') + 
  # geom_text(aes(label=MGE_per_genome_round),position=position_dodge(width=0.9), vjust=-0.25) + 
  scale_x_discrete(limits=c('insertion sequence','unit transposon','composite transposon','mite','ice'),
                   labels=c('IS','Unit Transposon','Composite Transposon','MITE', 'ICE')) +
  ylim2(limits = c(0,10)) + xlab('Mobile element type') + ylab(NULL) + labs(title = 'S. maltophilia') +
  theme_bw()  +
  theme(axis.ticks.y =element_blank(), axis.text.y =element_blank(),axis.text.x = element_text(angle = 30,hjust = 1,size=30)) +
  theme(strip.background = element_blank(),
        plot.title  = element_textbox(
          size = 34,
          color = "#4d4d4d", fill = "#dbdad9", box.color = "black",face = 'italic',
          halign = 0.5, linetype = 1,  width = unit(1, "npc"),
          margin = margin(0, 0, 0, 0),padding = margin(6, 6, 6, 6)),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.x = element_blank(),
        axis.title.x = element_text(size=36)) # change xlabs fontsize, xlab() --> axis.title.x()

df_statistics_EMHOS <- subset(df_statistics,species=='EHMOS')
p_EMHOS <- ggplot(df_statistics_EMHOS, aes(x=mge_type, y=MGE_per_genome)) +
  geom_errorbar(limits, width=.2,col='#b54030',
                position=position_dodge(.9)) +
  geom_bar(stat = "identity",fill='#b54030') + 
  # geom_text(aes(label=MGE_per_genome_round),position=position_dodge(width=0.9), vjust=-0.25) + 
  scale_x_discrete(limits=c('insertion sequence','unit transposon','composite transposon','mite','ice'),
                   labels=c('IS','Unit Transposon','Composite Transposon','MITE', 'ICE')) +
  ylim2(limits = c(0,10)) + xlab(NULL) + ylab(NULL) + labs(title = 'EHMOS') +
  theme_bw()  +
  theme(axis.ticks.y =element_blank(), axis.text.y =element_blank(),axis.text.x = element_text(angle = 30,hjust = 1,size=30)) +
  theme(strip.background = element_blank(),
        plot.title  = element_textbox(
          size = 34,
          color = "#4d4d4d", fill = "#dbdad9", box.color = "black",face = 'italic',
          halign = 0.5, linetype = 1,  width = unit(1, "npc"),
          margin = margin(0, 0, 0, 0),padding = margin(6, 6, 6, 6)),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.x = element_blank())

p_combined <- plot_list( p_acinetobacter, p_corynebacterium, p_klebsiella,p_pseudomonas,p_stenotrophomonas,p_EMHOS)
p_combined

```

```{r Fig 4b}

df_mge_abundance = read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Main_figures\mobile_elements_finder-stats_comparisons-mge_abundance.csv)")
df_mge_abundance$species <- factor(df_mge_abundance$species,levels = c('EHMOS','corynebacterium','stenotrophomonas','pseudomonas','acinetobacter','klebsiella'))
my_comparisons <- list(c('EHMOS','corynebacterium'),c('EHMOS','stenotrophomonas'),c('EHMOS','pseudomonas'),
                       c('EHMOS','acinetobacter'),c('EHMOS','klebsiella'))

ggboxplot(df_mge_abundance, x="species", y="mge_num",
          color = "species", palette =c("#E41A1C","#377EB8","#4DAF4A","#984EA3","#FF7F00","#F781BF","#A65628","#F781BF"),
          add = "jitter",add.params = list(size = 1.5, alpha = 0.5))  +
  stat_boxplot(mapping = aes(color=species),geom = "errorbar",width=0.1)  +
  stat_compare_means(comparisons = my_comparisons,label = "p.label",size=8,
                     vjust=.15,tip.length= c(0,0,0)) + 
  scale_x_discrete(labels=c('EHMOS','C. striatum','S. maltophilia','P. aeruginosa','A. baumannii','K. pneumoniae'))+
  labs(x=NULL,y='Number of mobile elements',) +  
  theme_pubr(base_size = 24,legend ='none',x.text.angle = 30) 

```

```{r Fig 4c}


df_profile <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Main_figures\mobile_elements_finder-profile.csv)')
df_profile <- subset(df_profile,species !='assembled_contigs_all')

# reorder the species column
df_profile$species <- factor(df_profile$species, levels=c("acinetobacter", "corynebacterium", "klebsiella","pseudomonas","stenotrophomonas","EHMOS"))
# change the facted plots labels
species.labs <- c("A. baumannii", "C. striatum", "K. pneumoniae", "P. aeruginosa", "S. maltophilia", "EHMOS")
names(species.labs) <- c("acinetobacter", "corynebacterium", "klebsiella","pseudomonas","stenotrophomonas","EHMOS")

p_profile <- ggplot(df_profile, aes(x=allele_len,fill=species)) + 
  geom_histogram(color="#e9ecef", alpha=1, position = 'identity') + 
  scale_x_log10(limits=c(100,100000),labels=scales::comma) + ylim2(limits = c(-1,40)) + 
  scale_fill_manual(values = c("acinetobacter"="#d08b70", "corynebacterium"='#988b74', "klebsiella"='#6fa9c8',
                               "pseudomonas"='#98c5b4',"stenotrophomonas"='#384073',"EHMOS"='#b54030')) +
  theme_bw() + 
  facet_grid(species ~ .,
             labeller = labeller(species = species.labs)) +
  xlab('Element length(bp)') + ylab('Number of mobile elements') +
  theme(axis.text.x  = element_text(size = 30),axis.text.y = element_text(size = 30),
        axis.title.x = element_text(size = 32),axis.title.y = element_text(size = 32)) +
  theme(strip.background = element_blank(),
        strip.text.y = element_textbox_simple(size = 22, # left y-label fontsize
                                              color = "#4d4d4d", fill = "#dbdad9", box.color = "black",face = 'italic', orientation = "right-rotated",
                                              halign = 0.5, linetype = 1,  width = unit(1, "npc"),
                                              margin = margin(0, 0, 0, 0),padding = margin(6, 6, 6, 6)),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.x = element_blank(),
        legend.position = 'none',panel.spacing.y = unit(2, "lines"))
p_profile


### claculate the p-values using Kolmogorov-Smirnov Test

vec <- c('pseudomonas','acinetobacter','klebsiella','corynebacterium','stenotrophomonas','EHMOS')

# Generate all pairs
pairs <- combn(vec, 2)

# Convert to a list of vectors, which can be used in stat_compare_means
comparison_list <- split(pairs, col(pairs))


# Conduct KS test for each pair
result_list <- lapply(comparison_list, function(pair) {
  group1 <- df_profile[df_profile$species == pair[1], "allele_len"]
  group2 <- df_profile[df_profile$species == pair[2], "allele_len"]
  ks_result <- ks.test(group1, group2)
  
  # Return a data frame
  data.frame(
    group1 = pair[1],
    group2 = pair[2],
    p_value = ks_result$p.value
  )
})

# Combine all results into one data frame
result_df <- do.call(rbind, result_list)

# r$> result_df
#              group1           group2      p_value
# 1       pseudomonas    acinetobacter 1.486184e-03
# 2       pseudomonas       klebsiella 2.595425e-03
# 3       pseudomonas  corynebacterium 3.880953e-02
# 4       pseudomonas stenotrophomonas 2.400837e-01
# 5       pseudomonas            EHMOS 6.234450e-05
# 6     acinetobacter       klebsiella 4.514242e-03
# 7     acinetobacter  corynebacterium 3.178381e-08
# 8     acinetobacter stenotrophomonas 7.544480e-02
# 9     acinetobacter            EHMOS 8.663095e-04
# 10       klebsiella  corynebacterium 2.106382e-03
# 11       klebsiella stenotrophomonas 1.432543e-01
# 12       klebsiella            EHMOS 8.003375e-03
# 13  corynebacterium stenotrophomonas 3.150332e-02
# 14  corynebacterium            EHMOS 1.700597e-03
# 15 stenotrophomonas            EHMOS 1.631064e-02

```

```{r Fig 4d}

# Sample dataset
nodes <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Main_figures\mobile_elements_finder-draft_genomes-arg_sankey-nodes.csv)',header = F) # save in the same file 'mobile_elements_finder-draft_genomes-arg_sankey.txt'
names(nodes) <- c('name')
links <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Main_figures\mobile_elements_finder-draft_genomes-arg_sankey-links-scaled.csv)') # save in the same file 'mobile_elements_finder-draft_genomes-arg_sankey.txt'

# Create Sankey plot using plotly
p <-plot_ly(type = "sankey",
        domain = list(
          x = c(0, 1),
          y = c(0, 1)
        ),
        orientation = "h",
        valueformat = ".0f",
        node = list(
          pad = 0,
          thickness = 20,
          line = list(
            color = "black",
            width = 0.5
          ),
          label = nodes$name
        ),
        link = list(
          source = links$source,
          target = links$target,
          value = links$normalized_value,
          color = c("rgba(204, 204, 204, 0.35)")
        ),
        width =1500,height =700) %>%
  layout(
    title = NULL,
    font = list(size = 32,family="Arial"),
    xaxis = list(showgrid = FALSE, zeroline = FALSE),
    yaxis = list(showgrid = FALSE, zeroline = FALSE)
  )

```

```{r Fig 4f}

### Custom function to generate the desired format of the matrix
###-------------------------------------------------------
#Get lower triangle of the correlation matrix
get_lower_tri<-function(cormat){
  cormat[upper.tri(cormat,diag = T)] <- NA
  return(cormat)
}
#Get upper triangle of the correlation matrix
get_upper_tri <- function(cormat){
  cormat[lower.tri(cormat,diag = T)]<- NA
  return(cormat)
}
###-------------------------------------------------------

df_plasmids_ani <- read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Main_figures\plasmids_selected-ANI-NZ_CP010782.1_fraglen.csv)")
#remove suffix: '.fasta' 
colnames(df_plasmids_ani) <- str_remove(colnames(df_plasmids_ani),'.fasta')
row.names(df_plasmids_ani) <- str_remove(df_plasmids_ani$query,'.fasta')

#dataframe --> martix
df_plasmids_ani <- df_plasmids_ani[,-1]
matrix_plasmids_ani <- as.matrix(df_plasmids_ani)

#Remeber not to modify the order of samples name, just copy the output of 'table(df$var1)' and 'table(df$var2)' !!!
x_axis<-c("NZ_CP010782.1",
          "NZ_CP008708.1",
          "NZ_CP008850.1",
          "NZ_CP020576.1",
          "NZ_CP030084.1",
          "NZ_CP050414.1",
          "NZ_CP066231.1")
y_axis<-rev(c("NZ_CP008708.1",
              "NZ_CP008850.1",
              "NZ_CP020576.1",
              "NZ_CP030084.1",
              "NZ_CP050414.1",
              "NZ_CP066231.1",
              "Patient_6"))

#reshape data: 'wide' --> 'long'
df <- melt(get_lower_tri(matrix_plasmids_ani),na.rm = T)
colnames(df) <- c('var1','var2','ANI')

#generate the order of "x_axis" and "y_axis"
table(df$var1) 
table(df$var2)

#custom the order so that the plot will display in a lower-left layout
df<-df %>%
  mutate(var1=factor(var1,levels = y_axis),
         var2=factor(var2,levels = x_axis))

#labels to display on the heatmap
txt.df<-data.frame(x=1:8,
                   y=8:1,
                   label=c(
                     "NZ_CP010782.1",
                     "NZ_CP008708.1",
                     "NZ_CP008850.1",
                     "NZ_CP020576.1",
                     "NZ_CP030084.1",
                     "NZ_CP050414.1",
                     "NZ_CP066231.1",
                     'Patient_6'))

df$ANI <- round(df$ANI,3)
p1 <- ggplot(data=df,aes(x=var2,y=var1))+
  geom_tile(aes(fill=ANI),
            color="black", width = 0.92, height = 0.92)+
  geom_text(aes(label=ANI),size=10) +
  geom_text(data=txt.df,
            aes(x=x,y=y,label=label),size=10,
            nudge_x = 0.45,nudge_y = -0.2)+
  scale_fill_gradient2(low = "#fcbcce",mid = '#f2f2f2',high = "#6cb7d5",midpoint = 99.95) +
  theme_bw()+
  theme(axis.text = element_blank(),
        axis.ticks = element_blank(),
        panel.grid = element_blank(),
        panel.border = element_blank(),
        legend.position = "right",
        legend.title = element_text(size = 24),
        legend.text = element_text(size = 24),
        axis.title = element_blank())+
  coord_cartesian(xlim = c(0,8),y=c(0,8)) +
  guides(fill=guide_colorbar(barheight = 10,
                             ticks.colour = "black"))

```

```{r Fig 4g}

bbone <- read_mauve_backbone(r'(LRT_microbiomes_critical_illness\Data_repository\Main_figures\plasmids_selected.backbone)',
                             ref=1, filter_low=0)
names <- c('NZ_CP010782.1',
           'Patient_6')
names(bbone$dna_segs) <- names


#Calculating the lengths by adding the length on both sides of the comparisons
for (i in 1:length(bbone$comparisons)){
  cmp <- bbone$comparisons[[i]]
  bbone$comparisons[[i]]$length <-
    abs(cmp$end1 - cmp$start1) + abs(cmp$end2 - cmp$start2)
}
plot_gene_map(dna_segs=bbone$dna_segs, 
              comparisons=bbone$comparisons,dna_seg_label_col = "black",
              global_color_scheme=c("length", "auto", "blue_red",0.9),
              override_color_schemes=TRUE)

```
### Figure S4
### --------------------------------------------------------------------------------------------------------------------------------------------

```{r Fig s4a}

# matrix_rgi_species_all_norm <- normalize(matrix_rgi_species_all,margin = 1,method = 'range',range = c(0,2))
matrix_rgi_species_all_norm <- normalize(matrix_rgi_species_all,margin = 1,method = 'scale')
colnames(matrix_rgi_species_all_norm) <- colnames(matrix_rgi_species_all)

plot_heatmap_species_all <- Heatmap(matrix_rgi_species_all_norm,
                                    #breaks = c(0,1,3),
                                    col = c('#ffffff','#ff0000'),
                                    # col = c('#f7faff','#08306b'),
                                    left_annotation  = rowAnnotation(df=labels_annotation_col,
                                                                     col=list(species=c(corynebacterium="#e36255", stenotrophomonas="#ec9a86",
                                                                                        pseudomonas="#a2c5c9", klebsiella="#f3c262", 
                                                                                        acinetobacter="#f1e0ce"))),
                                    #                                     col = list(category=c(core='#16519f',accessory='#f07e74'))),
                                    # row_split = factor(labels_annotation_col[,1],levels = c("corynebacterium",'stenotrophomonas','klebsiella','pseudomonas',"acinetobacter")),
                                    # row_split = c(30,9,12,13,27),
                                    #border_color = NA,
                                    cluster_rows =as.dendrogram(dend_species_all), cluster_columns =T,
                                    # cluster_rows = T,cluster_columns =T,
                                    show_column_dend = F,show_row_dend = T,rect_gp = gpar(col= "grey"),
                                    show_column_names = T,show_row_names = T)

```

```{r Fig s4b}

df_args_vs_vfs <-
  read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\spearman_correlations-numberf_of_ARGs+VFs.csv)')

ggscatterstats(
  data = df_args_vs_vfs,
  x = arg_num,
  y = vf_num,
  xlab = 'Number of ARGs',
  ylab = 'Number of VFs',
  bf.message = F,results.subtitle = F
)


cor.test(df_args_vs_vfs$arg_num,df_args_vs_vfs$vf_num,
         method = 'spearman')
# Spearman's rank correlation rho
# 
# data:  df_args_vs_vfs$arg_num and df_args_vs_vfs$vf_num
# S = 13349, p-value < 2.2e-16
# alternative hypothesis: true rho is not equal to 0
# sample estimates:
#       rho 
# 0.8936982 

```

```{r Fig s4c}

df_rgi_profile_acinetobacter = read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\RGI_profile-PCoA_acinetobacter.csv)", header= TRUE)
abundance_acinetobacter = df_rgi_profile_acinetobacter[,2:(ncol(df_rgi_profile_acinetobacter)-2)]
m_abundance_acinetobacter = as.matrix(abundance_acinetobacter)
diss_acinetobacter <- vegdist(m_abundance_acinetobacter, method = "bray")

# Perform PCoA using cmdscale()
pcoa_acinetobacter <- cmdscale(diss_acinetobacter, eig = TRUE, k = 2) 

#extract PCoA scores (PCoA1, PCoA2)
data_scores_acinetobacter = as.data.frame(pcoa_acinetobacter$points)
names(data_scores_acinetobacter) <- c("PCoA1", "PCoA2")

data_scores_acinetobacter$filename = df_rgi_profile_acinetobacter$filename
data_scores_acinetobacter$SamplingSite = df_rgi_profile_acinetobacter$SamplingSite

set.seed(123)
adonis2(m_abundance_acinetobacter ~ SamplingSite, data_scores_acinetobacter, method = "bray", permutations = 999)
# Permutation test for adonis under reduced model
# Terms added sequentially (first to last)
# Permutation: free
# Number of permutations: 999
# 
# adonis2(formula = m_abundance_acinetobacter ~ SamplingSite, data = data_scores_acinetobacter, permutations = 999, method = "bray")
# Df SumOfSqs      R2       F Pr(>F)
# SamplingSite  1 -0.00419 -0.0098 -0.2425  0.993
# Residual     25  0.43142  1.0098               
# Total        26  0.42724  1.0000   

p_scatter_acinetobacter <- ggscatter(data_scores_acinetobacter,
                                     x = "PCoA1", 
                                     y = "PCoA2",
                                     color = 'SamplingSite',
                                     palette = c("HospitalA"="#e64833","HospitalB"="#7f59af","HospitalC"="#3e5386"),
                                     #shape = "SamplingSite",
                                     size=5,
                                     ggtheme = theme_pubr(border = T,base_size = 45),
                                     legend = 'none',
                                     ellipse = T,ellipse.level = 0.90)  + ggtitle('A. baumannii') +
  ggeasy::easy_center_title() + theme(plot.title = element_text(face = "italic",size = 45))




###-----------------------------------------------------------------------------------------------------------------------------------------

###plot RGI profile for corynebacterium based on sampling site
df_rgi_profile_corynebacterium = read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\RGI_profile-PCoA_corynebacterium.csv)", header= TRUE)
abundance_corynebacterium = df_rgi_profile_corynebacterium[,2:(ncol(df_rgi_profile_corynebacterium)-2)]
m_abundance_corynebacterium = as.matrix(abundance_corynebacterium)


diss_corynebacterium <- vegdist(m_abundance_corynebacterium, method = "bray")
# Perform PCoA using cmdscale()
pcoa_corynebacterium <- cmdscale(diss_corynebacterium, eig = TRUE, k = 2) 

#extract PCoA scores (PCoA1, PCoA2)
data_scores_corynebacterium = as.data.frame(pcoa_corynebacterium$points)
names(data_scores_corynebacterium) <- c("PCoA1", "PCoA2")

data_scores_corynebacterium$filename = df_rgi_profile_corynebacterium$filename
data_scores_corynebacterium$SamplingSite = df_rgi_profile_corynebacterium$SamplingSite

set.seed(123)
adonis2(m_abundance_corynebacterium ~ SamplingSite, data_scores_corynebacterium, method = "bray", permutations = 999)
# Permutation test for adonis under reduced model
# Terms added sequentially (first to last)
# Permutation: free
# Number of permutations: 999
# 
# adonis2(formula = m_abundance_corynebacterium ~ SamplingSite, data = data_scores_corynebacterium, permutations = 999, method = "bray")
# Df SumOfSqs      R2      F Pr(>F)
# SamplingSite  1  0.15791 0.05235 1.5466  0.213
# Residual     28  2.85874 0.94765              
# Total        29  3.01665 1.00000 

p_scatter_corynebacterium <- ggscatter(data_scores_corynebacterium,
                                       x = "PCoA1", 
                                       y = "PCoA2",
                                       color = 'SamplingSite',
                                       palette = c("HospitalA"="#e64833","HospitalB"="#7f59af","HospitalC"="#3e5386"),
                                       #shape = "SamplingSite",
                                       size=5,
                                       ggtheme = theme_pubr(border = T,base_size = 45),
                                       legend = 'none',
                                       ellipse = T,ellipse.level = 0.90)  + ggtitle('C. striatum') +
  ggeasy::easy_center_title() + theme(plot.title = element_text(face = "italic",size = 45))

###-----------------------------------------------------------------------------------------------------------------------------------------

###plot RGI profile for klebsiella based on sampling site
df_rgi_profile_klebsiella = read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\RGI_profile-PCoA_klebsiella.csv)", header= TRUE)
abundance_klebsiella = df_rgi_profile_klebsiella[,2:(ncol(df_rgi_profile_klebsiella)-2)]
m_abundance_klebsiella = as.matrix(abundance_klebsiella)

diss_klebsiella <- vegdist(m_abundance_klebsiella, method = "bray")

# Perform PCoA using cmdscale()
pcoa_klebsiella <- cmdscale(diss_klebsiella, eig = TRUE, k = 2) 

#extract PCoA scores (PCoA1, PCoA2)
data_scores_klebsiella = as.data.frame(pcoa_klebsiella$points)
names(data_scores_klebsiella) <- c("PCoA1", "PCoA2")

data_scores_klebsiella$filename = df_rgi_profile_klebsiella$filename
data_scores_klebsiella$SamplingSite = df_rgi_profile_klebsiella$SamplingSite

set.seed(123)
adonis2(m_abundance_klebsiella ~ SamplingSite, data_scores_klebsiella, method = "bray", permutations = 999)
# Permutation test for adonis under reduced model
# Terms added sequentially (first to last)
# Permutation: free
# Number of permutations: 999
# 
# adonis2(formula = m_abundance_klebsiella ~ SamplingSite, data = data_scores_klebsiella, permutations = 999, method = "bray")
# Df SumOfSqs      R2      F Pr(>F)  
# SamplingSite  2  0.14675 0.40922 3.4634  0.022 *
#   Residual     10  0.21186 0.59078                
# Total        12  0.35861 1.00000                

p_scatter_klebsiella <- ggscatter(data_scores_klebsiella,
                                  x = "PCoA1", 
                                  y = "PCoA2",
                                  color = 'SamplingSite',
                                  palette = c("HospitalA"="#e64833","HospitalB"="#7f59af","HospitalC"="#3e5386"),
                                  #shape = "SamplingSite",
                                  size=5,
                                  ggtheme = theme_pubr(border = T,base_size = 45),
                                  legend = 'right',
                                  ellipse = T,ellipse.level = 0.90)  + ggtitle('K. pneumoniae') +
  ggeasy::easy_center_title() + theme(plot.title = element_text(face = "italic",size = 45))


###-----------------------------------------------------------------------------------------------------------------------------------------

###plot RGI profile for pseudomonas based on sampling site
df_rgi_profile_pseudomonas = read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\RGI_profile-PCoA_pseudomonas.csv)", header= TRUE)
abundance_pseudomonas = df_rgi_profile_pseudomonas[,2:(ncol(df_rgi_profile_pseudomonas)-2)]
m_abundance_pseudomonas = as.matrix(abundance_pseudomonas)

diss_pseudomonas <- vegdist(m_abundance_pseudomonas, method = "bray")

# Perform PCoA using cmdscale()
pcoa_pseudomonas <- cmdscale(diss_pseudomonas, eig = TRUE, k = 2) 

#extract PCoA scores (PCoA1, PCoA2)
data_scores_pseudomonas = as.data.frame(pcoa_pseudomonas$points)
names(data_scores_pseudomonas) <- c("PCoA1", "PCoA2")

data_scores_pseudomonas$filename = df_rgi_profile_pseudomonas$filename
data_scores_pseudomonas$SamplingSite = df_rgi_profile_pseudomonas$SamplingSite

set.seed(123)
adonis2(m_abundance_pseudomonas ~ SamplingSite, data_scores_pseudomonas, method = "bray", permutations = 999)
# Permutation test for adonis under reduced model
# Terms added sequentially (first to last)
# Permutation: free
# Number of permutations: 999
# 
# adonis2(formula = m_abundance_pseudomonas ~ SamplingSite, data = data_scores_pseudomonas, permutations = 999, method = "bray")
# Df SumOfSqs     R2      F Pr(>F)
# SamplingSite  2 0.029636 0.1546 0.8229  0.657
# Residual      9 0.162059 0.8454              
# Total        11 0.191696 1.0000 

p_scatter_pseudomonas <- ggscatter(data_scores_pseudomonas,
                                   x = "PCoA1", 
                                   y = "PCoA2",
                                   color = 'SamplingSite',
                                   palette = c("HospitalA"="#e64833","HospitalB"="#7f59af","HospitalC"="#3e5386"),
                                   #shape = "SamplingSite",
                                   size=5,
                                   ggtheme = theme_pubr(border = T,base_size = 45),
                                   legend = 'none',
                                   ellipse = T,ellipse.level = 0.90)  + ggtitle('P. aeruginosa') +
  ggeasy::easy_center_title() + theme(plot.title = element_text(face = "italic",size = 45))

###-----------------------------------------------------------------------------------------------------------------------------------------

###plot RGI profile for stenotrophomonas based on sampling site
df_rgi_profile_stenotrophomonas = read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\RGI_profile-PCoA_stenotrophomonas.csv)", header= TRUE)
abundance_stenotrophomonas = df_rgi_profile_stenotrophomonas[,2:(ncol(df_rgi_profile_stenotrophomonas)-2)]
m_abundance_stenotrophomonas = as.matrix(abundance_stenotrophomonas)

diss_stenotrophomonas <- vegdist(m_abundance_stenotrophomonas, method = "bray")

# Perform PCoA using cmdscale()
pcoa_stenotrophomonas <- cmdscale(diss_stenotrophomonas, eig = TRUE, k = 2) 

#extract PCoA scores (PCoA1, PCoA2)
data_scores_stenotrophomonas = as.data.frame(pcoa_stenotrophomonas$points)
names(data_scores_stenotrophomonas) <- c("PCoA1", "PCoA2")


data_scores_stenotrophomonas$filename = df_rgi_profile_stenotrophomonas$filename
data_scores_stenotrophomonas$SamplingSite = df_rgi_profile_stenotrophomonas$SamplingSite


set.seed(123)
adonis2(m_abundance_stenotrophomonas ~ SamplingSite, data_scores_stenotrophomonas, method = "bray", permutations = 999)
# Permutation test for adonis under reduced model
# Terms added sequentially (first to last)
# Permutation: free
# Number of permutations: 999
# 
# adonis2(formula = m_abundance_stenotrophomonas ~ SamplingSite, data = data_scores_stenotrophomonas, permutations = 999, method = "bray")
# Df SumOfSqs      R2      F Pr(>F)
# SamplingSite  1  0.11071 0.15611 1.2949  0.263
# Residual      7  0.59850 0.84389              
# Total         8  0.70922 1.00000 


p_scatter_stenotrophomonas <- ggscatter(data_scores_stenotrophomonas,
                                        x = "PCoA1", 
                                        y = "PCoA2",
                                        color = 'SamplingSite',
                                        palette = c("HospitalA"="#e64833","HospitalB"="#7f59af","HospitalC"="#3e5386"),
                                        #shape = "SamplingSite",
                                        size=5,
                                        ggtheme = theme_pubr(border = T,base_size = 45),
                                        legend = 'none',
                                        ellipse = T,ellipse.level = 0.90)  + ggtitle('S. maltophilia') +
  ggeasy::easy_center_title() + theme(plot.title = element_text(face = "italic",size = 45))


(p_scatter_acinetobacter|p_scatter_corynebacterium|p_scatter_klebsiella|p_scatter_pseudomonas|p_scatter_stenotrophomonas)

```

```{r Fig s4d}

df_vfdb_profile_acinetobacter = read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\vfdb_profile-PCoA_acinetobacter.csv)", header= TRUE)
abundance_acinetobacter = df_vfdb_profile_acinetobacter[,2:(ncol(df_vfdb_profile_acinetobacter)-2)]
m_abundance_acinetobacter = as.matrix(abundance_acinetobacter)


diss_acinetobacter <- vegdist(m_abundance_acinetobacter, method = "bray")
pcoa_acinetobacter <- cmdscale(diss_acinetobacter, eig = TRUE, k = 2) 

data_scores_acinetobacter = as.data.frame(pcoa_acinetobacter$points)
names(data_scores_acinetobacter) <- c("PCoA1", "PCoA2")

data_scores_acinetobacter$filename = df_vfdb_profile_acinetobacter$filename
data_scores_acinetobacter$SamplingSite = df_vfdb_profile_acinetobacter$SamplingSite

set.seed(123)
adonis2(m_abundance_acinetobacter ~ SamplingSite, data_scores_acinetobacter, method = "bray", permutations = 999)
# Permutation test for adonis under reduced model
# Terms added sequentially (first to last)
# Permutation: free
# Number of permutations: 999
# 
# adonis2(formula = m_abundance_acinetobacter ~ SamplingSite, data = data_scores_acinetobacter, permutations = 999, method = "bray")
# Df SumOfSqs      R2      F Pr(>F)
# SamplingSite  1  0.01803 0.04527 1.1854  0.315
# Residual     25  0.38015 0.95473              
# Total        26  0.39817 1.00000  

p_scatter_acinetobacter <- ggscatter(data_scores_acinetobacter,
                                     x = "PCoA1", 
                                     y = "PCoA2",
                                     color = 'SamplingSite',
                                     palette = c("HospitalA"="#e64833","HospitalB"="#7f59af","HospitalC"="#3e5386"),
                                     #shape = "SamplingSite",
                                     size=5,
                                     ggtheme = theme_pubr(border = T,base_size = 45),
                                     legend = 'none',
                                     ellipse = T,ellipse.level = 0.90)  + ggtitle('A. baumannii') +
  ggeasy::easy_center_title() + theme(plot.title = element_text(face = "italic",size = 45))




###-----------------------------------------------------------------------------------------------------------------------------------------

###plot vfdb profile for corynebacterium based on sampling site
df_vfdb_profile_corynebacterium = read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\vfdb_profile-PCoA_corynebacterium.csv)", header= TRUE)
abundance_corynebacterium = df_vfdb_profile_corynebacterium[,2:(ncol(df_vfdb_profile_corynebacterium)-2)]
m_abundance_corynebacterium = as.matrix(abundance_corynebacterium)


diss_corynebacterium <- vegdist(m_abundance_corynebacterium, method = "bray")
pcoa_corynebacterium <- cmdscale(diss_corynebacterium, eig = TRUE, k = 2) 

data_scores_corynebacterium = as.data.frame(pcoa_corynebacterium$points)
names(data_scores_corynebacterium) <- c("PCoA1", "PCoA2")

data_scores_corynebacterium$filename = df_vfdb_profile_corynebacterium$filename
data_scores_corynebacterium$SamplingSite = df_vfdb_profile_corynebacterium$SamplingSite

set.seed(123)
adonis2(m_abundance_corynebacterium ~ SamplingSite, data_scores_corynebacterium, method = "bray", permutations = 999)
# Permutation test for adonis under reduced model
# Terms added sequentially (first to last)
# Permutation: free
# Number of permutations: 999
# 
# adonis2(formula = m_abundance_corynebacterium ~ SamplingSite, data = data_scores_corynebacterium, permutations = 999, method = "bray")
# Df SumOfSqs      R2      F Pr(>F)
# SamplingSite  1   0.2660 0.04841 1.3228   0.23
# Residual     26   5.2276 0.95159              
# Total        27   5.4936 1.00000  

p_scatter_corynebacterium <- ggscatter(data_scores_corynebacterium,
                                       x = "PCoA1", 
                                       y = "PCoA2",
                                       color = 'SamplingSite',
                                       palette = c("HospitalA"="#e64833","HospitalB"="#7f59af","HospitalC"="#3e5386"),
                                       #shape = "SamplingSite",
                                       size=5,
                                       ggtheme = theme_pubr(border = T,base_size = 45),
                                       legend = 'none',
                                       ellipse = T,ellipse.level = 0.90)  + ggtitle('C. striatum') +
  ggeasy::easy_center_title() + theme(plot.title = element_text(face = "italic",size = 45))




###-----------------------------------------------------------------------------------------------------------------------------------------

###plot vfdb profile for klebsiella based on sampling site
df_vfdb_profile_klebsiella = read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\vfdb_profile-PCoA_klebsiella.csv)", header= TRUE)
abundance_klebsiella = df_vfdb_profile_klebsiella[,2:(ncol(df_vfdb_profile_klebsiella)-2)]
m_abundance_klebsiella = as.matrix(abundance_klebsiella)

diss_klebsiella <- vegdist(m_abundance_klebsiella, method = "bray")
pcoa_klebsiella <- cmdscale(diss_klebsiella, eig = TRUE, k = 2) 

data_scores_klebsiella = as.data.frame(pcoa_klebsiella$points)
names(data_scores_klebsiella) <- c("PCoA1", "PCoA2")

data_scores_klebsiella$filename = df_vfdb_profile_klebsiella$filename
data_scores_klebsiella$SamplingSite = df_vfdb_profile_klebsiella$SamplingSite

set.seed(123)
adonis2(m_abundance_klebsiella ~ SamplingSite, data_scores_klebsiella, method = "bray", permutations = 999)
# Permutation test for adonis under reduced model
# Terms added sequentially (first to last)
# Permutation: free
# Number of permutations: 999
# 
# adonis2(formula = m_abundance_klebsiella ~ SamplingSite, data = data_scores_klebsiella, permutations = 999, method = "bray")
# Df SumOfSqs      R2      F Pr(>F)  
# SamplingSite  2  0.51338 0.30725 2.2177   0.08 .
# Residual     10  1.15749 0.69275                
# Total        12  1.67087 1.00000

p_scatter_klebsiella <- ggscatter(data_scores_klebsiella,
                                  x = "PCoA1", 
                                  y = "PCoA2",
                                  color = 'SamplingSite',
                                  palette = c("HospitalA"="#e64833","HospitalB"="#7f59af","HospitalC"="#3e5386"),
                                  #shape = "SamplingSite",
                                  size=5,
                                  ggtheme = theme_pubr(border = T,base_size = 45),
                                  legend = 'none',
                                  ellipse = T,ellipse.level = 0.90)  + ggtitle('K. pneumoniae') +
  ggeasy::easy_center_title() + theme(plot.title = element_text(face = "italic",size = 45))




###-----------------------------------------------------------------------------------------------------------------------------------------

###plot vfdb profile for pseudomonas based on sampling site
df_vfdb_profile_pseudomonas = read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\vfdb_profile-PCoA_pseudomonas.csv)", header= TRUE)
abundance_pseudomonas = df_vfdb_profile_pseudomonas[,2:(ncol(df_vfdb_profile_pseudomonas)-2)]
m_abundance_pseudomonas = as.matrix(abundance_pseudomonas)


diss_pseudomonas <- vegdist(m_abundance_pseudomonas, method = "bray")
pcoa_pseudomonas <- cmdscale(diss_pseudomonas, eig = TRUE, k = 2) 

data_scores_pseudomonas = as.data.frame(pcoa_pseudomonas$points)
names(data_scores_pseudomonas) <- c("PCoA1", "PCoA2")

data_scores_pseudomonas$filename = df_vfdb_profile_pseudomonas$filename
data_scores_pseudomonas$SamplingSite = df_vfdb_profile_pseudomonas$SamplingSite

set.seed(123)
adonis2(m_abundance_pseudomonas ~ SamplingSite, data_scores_pseudomonas, method = "bray", permutations = 999)
# Permutation test for adonis under reduced model
# Terms added sequentially (first to last)
# Permutation: free
# Number of permutations: 999
# 
# adonis2(formula = m_abundance_pseudomonas ~ SamplingSite, data = data_scores_pseudomonas, permutations = 999, method = "bray")
# Df SumOfSqs      R2      F Pr(>F)
# SamplingSite  2  0.11810 0.10734 0.5411  0.846
# Residual      9  0.98211 0.89266              
# Total        11  1.10021 1.00000 


p_scatter_pseudomonas <- ggscatter(data_scores_pseudomonas,
                                   x = "PCoA1", 
                                   y = "PCoA2",
                                   color = 'SamplingSite',
                                   palette = c("HospitalA"="#e64833","HospitalB"="#7f59af","HospitalC"="#3e5386"),
                                   #shape = "SamplingSite",
                                   size=5,
                                   ggtheme = theme_pubr(border = T,base_size = 45),
                                   legend = 'none',
                                   ellipse = T,ellipse.level = 0.90)  + ggtitle('P. aeruginosa') +
  ggeasy::easy_center_title() + theme(plot.title = element_text(face = "italic",size = 45))


###-----------------------------------------------------------------------------------------------------------------------------------------

###plot vfdb profile for stenotrophomonas based on sampling site
df_vfdb_profile_stenotrophomonas = read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\vfdb_profile-PCoA_stenotrophomonas.csv)", header= TRUE)
abundance_stenotrophomonas = df_vfdb_profile_stenotrophomonas[,2:(ncol(df_vfdb_profile_stenotrophomonas)-2)]
m_abundance_stenotrophomonas = as.matrix(abundance_stenotrophomonas)


diss_stenotrophomonas <- vegdist(m_abundance_stenotrophomonas, method = "bray")
pcoa_stenotrophomonas <- cmdscale(diss_stenotrophomonas, eig = TRUE, k = 2) 

data_scores_stenotrophomonas = as.data.frame(pcoa_stenotrophomonas$points)
names(data_scores_stenotrophomonas) <- c("PCoA1", "PCoA2")

data_scores_stenotrophomonas$filename = df_vfdb_profile_stenotrophomonas$filename
data_scores_stenotrophomonas$SamplingSite = df_vfdb_profile_stenotrophomonas$SamplingSite


set.seed(123)
adonis2(m_abundance_stenotrophomonas ~ SamplingSite, data_scores_stenotrophomonas, method = "bray", permutations = 999)
# Permutation test for adonis under reduced model
# Terms added sequentially (first to last)
# Permutation: free
# Number of permutations: 999
# 
# adonis2(formula = m_abundance_stenotrophomonas ~ SamplingSite, data = data_scores_stenotrophomonas, permutations = 999, method = "bray")
# Df SumOfSqs      R2      F Pr(>F)
# SamplingSite  1 0.007856 0.06762 0.5076  0.481
# Residual      7 0.108335 0.93238              
# Total         8 0.116191 1.00000   


p_scatter_stenotrophomonas <- ggscatter(data_scores_stenotrophomonas,
                                        x = "PCoA1", 
                                        y = "PCoA2",
                                        color = 'SamplingSite',
                                        palette = c("HospitalA"="#e64833","HospitalB"="#7f59af","HospitalC"="#3e5386"),
                                        #shape = "SamplingSite",
                                        size=5,
                                        ggtheme = theme_pubr(border = T,base_size = 45),
                                        legend = 'none',
                                        ellipse = T,ellipse.level = 0.90)  + ggtitle('S. maltophilia') +
  ggeasy::easy_center_title() + theme(plot.title = element_text(face = "italic",size = 45))

(p_scatter_acinetobacter|p_scatter_corynebacterium|p_scatter_klebsiella|p_scatter_pseudomonas|p_scatter_stenotrophomonas)

```

### Figure 5
### --------------------------------------------------------------------------------------------------------------------------------------------

```{r Fig 5a}

df_snps_p  <- read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Main_figures\acinetobacter-snps_distribution-merged.csv)")
df_snps_p$position[df_snps_p$types =='synonymous_variant'] <- df_snps_p$position[df_snps_p$types =='synonymous_variant'] - 5000
df_snps_p$type <- gsub("Style", "Pattern ", df_snps_p$type)

p_snps_distribution <- ggplot(df_snps_p) + geom_segment(aes(x=position,y=0,xend=position,yend=snps_counts,color=types),data=df_snps_p,size=.3)  +
  scale_color_manual(values=c('nonsynonymous_variant'='red','synonymous_variant'='blue'),
                     name='',
                     labels = c('Nonsynonymous','Synonymous')) + 
  scale_x_continuous(labels= scales::comma)  +
  guides(color=guide_legend(nrow = 2)) + facet_grid(type~.,scales = "free_y") +
  ggtitle('Acinetobacter baumannii')

ggpar(p_snps_distribution,
      xlab = "Genomic position",
      ylab = "SNP number") + theme_pubr(base_size = 20,legend = "none",margin = FALSE,border = T) + 
  ggeasy::easy_center_title() + theme(plot.title = element_text(face = "italic"))

```

```{r Fig 5b}

df_snps_p  <- read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Main_figures\klebsiella-snps_distribution-merged.csv)")
df_snps_p$position[df_snps_p$types =='synonymous_variant'] <- df_snps_p$position[df_snps_p$types =='synonymous_variant'] - 5000
df_snps_p$type <- gsub("Style", "Pattern ", df_snps_p$type)

p_snps_distribution <- ggplot(df_snps_p) + geom_segment(aes(x=position,y=0,xend=position,yend=snps_counts,color=types),data=df_snps_p,size=.3)  +
  scale_color_manual(values=c('nonsynonymous_variant'='red','synonymous_variant'='blue'),
                     name='',
                     labels = c('Nonsynonymous','Synonymous')) + 
  scale_x_continuous(labels= scales::comma)  +
  guides(color=guide_legend(nrow = 2)) + facet_grid(type~.,scales = "free_y") +
  ggtitle('Klebsiella pneumoniae') 

ggpar(p_snps_distribution,
      xlab = "Genomic position",
      ylab = "SNP number") + theme_pubr(base_size = 20,legend = "none",margin = FALSE,border = T) + 
  ggeasy::easy_center_title() + theme(plot.title = element_text(face = "italic"))

```

```{r Fig 5c,d}

df_recombinations_all <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Main_figures\acinetobacter-recombinations-consensus_blocks_all.csv)') # save in the same file 'acinetobacter-recombination.txt'
df_pi <- read.table(r"(LRT_microbiomes_critical_illness\Data_repository\Main_figures\all_patients-acinetobacter.pi)",header = T) # save in the same file 'acinetobacter-recombination.txt'


p_recombinations <- ggplot() + 
  geom_segment(aes(x=coordinates,y=y_start,xend=coordinates,yend=y_end,size=cds_length),
               data=df_recombinations_all,color='#377eb8') +
  labs(x='Position on Chromosome',y='Recombination\nblocks') + 
  scale_x_continuous(expand = c(0.02, 0),limits = c(0,4100000),labels= scales::comma) +
  scale_y_continuous(expand = c(0, 0)) +
  theme_bw(base_size = 35) +
  theme(axis.text.y = element_blank(),
        axis.ticks.y = element_blank(),legend.position = 'none')

p_pi<-ggplot(df_pi,aes(x = BIN_END, y = PI))+
  geom_line(size = 0.66,color='#e41a1c') +
  theme_classic2()+
  theme(panel.spacing = unit(1.5, "cm"),
        strip.background = element_blank(),
        strip.placement = "outside",
        legend.position = "none",
        plot.margin=unit(c(1,1,1,1), "cm"))+
  scale_x_continuous(expand = c(0.02, 0),limits = c(0,4100000),labels= scales::comma) +
  scale_y_continuous(expand = c(0, 0)) + theme_bw(base_size = 35)
p_pi <- ggpar(p_pi,
              xlab = "",
              ylab= "Pi") 

ggarrange(p_pi,p_recombinations,ncol=1,nrow=2,align = c("hv")) + theme(plot.margin = margin(0.1,1.6,2,0.1, "cm")) 


```

```{r Fig 5e}

my_order <- c('P151','P156','P105','P132','P114','P141','P124','P112','P157','P144','P26','P23',
              'P146','P52','P77','P136','P25','P123','P143','P130','P118','P113')
my_palette <- c("#267278","#65338D","#4770B3","#D21F75","#3B3689","#50AED3","#48B24F",
                "#E57438","#569DD2","#569D79","#58595B","#84D2F4","#CAD93f","#F5C8AF","#9AC483","#9E9EA2",
                "#F2F4F9","#51A2DA","#B8D1FF","#C2CEE5", "#F0F921FF", "#FCA636FF")
my_cols <- c("#0D0887FF", "#6A00A8FF", "#B12A90FF",
             "#E16462FF", "#FCA636FF", "#F0F921FF")

df_acinetobacter <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Main_figures\all_patients-snps_distribution-acinetobacter-region1.csv)')
df_acinetobacter_filtered <- subset(df_acinetobacter,PatientID %in% my_order)
df_tree2 <- read.tree(r"(LRT_microbiomes_critical_illness\Data_repository\Main_figures\draft_only-snpscalling_filtered_resolved.tre)")
tree_order_sampleid <- c('P105','P124','P114','P156','P141','P132','P151','P118','P130','P113','P143',
                         'P123','P144','P112','P136','P77','P25','P52','P146','P23','P26','P157')
df_acinetobacter_filtered2 <- subset(df_acinetobacter,PatientID %in% tree_order_sampleid)

## plot ballonplot with ggplot2 directly
plot_ballon <- ggplot(df_acinetobacter_filtered2,aes(x=LOCUS_TAG,y=PatientID))+
  geom_point(aes(size=`percentage`,
                 color=`percentage`))+
  theme_bw()+
  theme(panel.grid = element_blank(),
        axis.text.x=element_text(angle=90,vjust=0.5,hjust = 1))+
  labs(y=NULL) +
  scale_color_gradientn(colors =rev(brewer.pal(n = 10, name = 'RdBu'))) +
  theme(axis.text.x = element_text(size=18,color='black'),axis.text.y = element_text(size=18,color='black'),
        axis.title = element_text(size=20)) +
  guides(size=guide_legend(order=3))

plot_tree <- ggtree(df_tree2,branch.length = 'none') 
plot_ballon %>%
  insert_left(plot_tree,width =0.2) 

```

```{r Fig 5e}

### the strcuture of the putative phage was ploted with python

from dna_features_viewer import GraphicFeature,BiopythonTranslator,GraphicRecord
import matplotlib.      pyplot as plt
import matplotlib
color_map = {
    "rep_origin": "yellow",
    "CDS": "blue",
    "regulatory": "red",
    "misc_recomb": "darkblue",
    "misc_feature": "lightblue",
}

def feature_properties(f):

    return dict(fontdict={'size':32},box_linewidth=1.2,color='#4dbad4')


translator = BiopythonTranslator(
    features_filters=(lambda f: f.type not in ["gene"],),
    features_properties=feature_properties
)

graphic_record = translator.translate_record(r"LRT_microbiomes_critical_illness\Data_repository\Main_figures\GCA_025677765.1_subsets_snps_phage.gb")
graphic_record_sub = graphic_record.crop((3317571,3330094))
matplotlib.rcdefaults()
plt.rcParams.update({'font.size': 32})
ax, _ = graphic_record_sub.plot(figure_width=40,figure_height=12,strand_in_label_threshold=7,max_label_length=50,max_line_length=30,level_offset=2.4,elevate_outline_annotations=False)
ax.figure.tight_layout()

```

### Figure S5
### --------------------------------------------------------------------------------------------------------------------------------------------

```{r Fig s5a}

df_top_abundant_mges <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\mobile_elements_finder-contigs_all-most_abundant_mges.csv)')

plot_top_abundant_mges<- ggbarplot(df_top_abundant_mges[1:6,],x = "name", y = "counts",#orientation = "horiz",
                                  fill = "#a6cee3", color = "#a6cee3") + theme_bw(base_size = 24) +
  theme(axis.text.x = element_text(color = 'black'),
        axis.text.y = element_text(color = 'black'))

ggpar(plot_top_abundant_mges,
               xlab = "Mobile element",
               ylab = "Number of\nmobile elements",
      x.text.angle = 30)  

```

```{r Fig s5b}

df_mash_screen_arg <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\mash_screen-MGEs-carrying_ARG-contigs_allarg_prevalence.csv)')


plot_mash_screen_arg <- ggbarplot(df_mash_screen_arg[12:1,],x = "Best_Hit_ARO", y = "counts",sort.val = 'desc',#orientation = "horiz",
                           fill = "#a6cee3", color = "#a6cee3") + theme_bw(base_size = 24)+
  theme(axis.text.x = element_text(color = 'black'),
        axis.text.y = element_text(color = 'black'))

ggpar(plot_mash_screen_arg,
      xlab = "Gene",
      ylab = "Number of\ntransferable ARGs",
      x.text.angle = 30) 


```

```{r Fig s5c}

df_prevalence_species_mobilome <- read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\prevalence_comparisons-five_prevalent_species+EHMOS-metaphlan.csv)")
plot_df <- ggplot(df_prevalence_species_mobilome, aes(fill=SamplingSite, x=reorder(taxonomy,prevalency_percentage,sum,decreasing=T), y=prevalency_percentage))+  #'x=reorder(species,prevalency_percentage,sum,decreasing=T)' --> order stacked barplot with sum of values
  geom_bar(position = "stack",stat = "identity",) +
  scale_fill_manual(values = c('Hospital-A'='#d93526','Hospital-B'='#f7a902','Hospital-C'='#1a6196'))+
  scale_x_discrete(labels=c('C. striatum','S. maltophilia','P. aeruginosa','K. pneumoniae','A. baumannii',
                            'E. anophelis','P. melaninogenica','P. oris','N. subflava','H. influenzae'))+
  theme_pubr(x.text.angle = 35)
plot_df


ggpar(plot_df,
      ylab = "Percentage of detected\nsamples (%)",
      xlab = "Species",
      font.xtickslab=c(18),
      font.ytickslab=c(18),
      font.x = c(18,"bold"),
      font.y = c(18,"bold"),
      font.legend = c(18,'bold'),
      x.text.angle =50) + theme_pubr(base_size = 30,x.text.angle =35,legend = "none",margin = FALSE)

```

```{r Fig s5d}

df_prevalence_species_mobilome <- read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\prevalence_comparisons-five_prevalent_species+EHMOS.csv)")

my_comparisons <- list(c('prevalent','EHMOS'))
ggboxplot(df_prevalence_species_mobilome, x="group", y="prevalency_percentage",fill = 'group',
          color = "group", palette =c("#E41A1C","#377EB8","#4DAF4A","#984EA3","#FF7F00","#F781BF","#A65628","#F781BF"),
          add = "jitter",add.params = list(size = 1.5, alpha = 0.5))  +
  stat_boxplot(mapping = aes(color=group),geom = "errorbar",width=0.1)  +
  stat_compare_means(comparisons = my_comparisons,label = "p.label",size=12,
                     vjust=.15,tip.length= c(0,0,0)) + 
  scale_x_discrete(labels=c('More prevalent species','EHMOS'))+
  labs(x=NULL,y='Percentage of detected\nsamples (%)',) +  
  theme_pubr(base_size = 36,legend ='none',x.text.angle = 30) 

```

```{r Fig s5e}

df_mge_abundance <- read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\mobile_elements_finder-stats_comparisons-mge_abundance.csv)")
my_comparisons <- list(c('prevalent','EHMOS'))

ggboxplot(df_mge_abundance, x="group", y="mge_num",
          color = "group", palette =c("#E41A1C","#377EB8","#4DAF4A","#984EA3","#FF7F00","#F781BF","#A65628","#F781BF"),
          add = "jitter",add.params = list(size = 1.5, alpha = 0.5))  +
  stat_boxplot(mapping = aes(color=group),geom = "errorbar",width=0.1)  +
  stat_compare_means(comparisons = my_comparisons,label = "p.label",size=12,
                     vjust=.15,tip.length= c(0,0,0)) + 
scale_x_discrete(labels=c('More prevalent species','EHMOS'))+
  labs(x=NULL,y='Number of mobile elements',) +  
  theme_pubr(base_size = 36,legend ='none',x.text.angle = 30) 

```

```{r Fig s5f,g}

### Fig s5g
bbone <- read_mauve_backbone(r'(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\plasmids_selected-comparisons-NZ_CP010782.1.backbone)', # save in the same file 'plasmids_selected-comparisons.backbone'
                             ref=1, filter_low=0)
names <- c('NZ_CP010782.1',
           'Patient_6')
names(bbone$dna_segs) <- names



#Calculating the lengths by adding the length on both sides of the comparisons
for (i in 1:length(bbone$comparisons)){
  cmp <- bbone$comparisons[[i]]
  bbone$comparisons[[i]]$length <-
    abs(cmp$end1 - cmp$start1) + abs(cmp$end2 - cmp$start2)
}
plot_gene_map(dna_segs=bbone$dna_segs, 
              comparisons=bbone$comparisons,dna_seg_label_col = "black",
              global_color_scheme=c("length", "auto", "blue_red",0.9),
              override_color_schemes=TRUE)

### Fig s5g
bbone <- read_mauve_backbone(r'(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\plasmids_selected-comparisons-NZ_CP045266.1.backbone)', # save in the same file 'plasmids_selected-comparisons.backbone'
                             ref=1, filter_low=0)
names <- c('NZ_CP030131.1',
           'NZ_CP058229.1',
           'NZ_CP045266.1',
           'NZ_CP028795.1',
           'NZ_CP034127.1',
           'NZ_CP066522.1',
           'NZ_CP069175.1',
           'Patient_115')
names(bbone$dna_segs) <- names



#Calculating the lengths by adding the length on both sides of the comparisons
for (i in 1:length(bbone$comparisons)){
  cmp <- bbone$comparisons[[i]]
  bbone$comparisons[[i]]$length <-
    abs(cmp$end1 - cmp$start1) + abs(cmp$end2 - cmp$start2)
}
plot_gene_map(dna_segs=bbone$dna_segs, 
              comparisons=bbone$comparisons,dna_seg_label_col = "black",
              global_color_scheme=c("length", "auto", "blue_red",0.9),
              override_color_schemes=TRUE)

```

```{r Fig s5h}
### Custom function to generate the desired format of the matrix
###-------------------------------------------------------
#Get lower triangle of the correlation matrix
get_lower_tri<-function(cormat){
  cormat[upper.tri(cormat,diag = T)] <- NA
  return(cormat)
}
#Get upper triangle of the correlation matrix
get_upper_tri <- function(cormat){
  cormat[lower.tri(cormat,diag = T)]<- NA
  return(cormat)
}
###-------------------------------------------------------

df_plasmids_ani <- read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\plasmids_selected-ANI-NZ_CP045266.1_fraglen.csv)")

#remove suffix: '.fasta' 
colnames(df_plasmids_ani) <- str_remove(colnames(df_plasmids_ani),'.fasta')
row.names(df_plasmids_ani) <- str_remove(df_plasmids_ani$query,'.fasta')

#dataframe --> martix
df_plasmids_ani <- df_plasmids_ani[,-1]
matrix_plasmids_ani <- as.matrix(df_plasmids_ani)

x_axis<-c("NZ_CP030131.1",
          "NZ_CP058229.1",
          "NZ_CP045266.1",
          "NZ_CP028795.1",
          "NZ_CP034127.1",
          "NZ_CP066522.1",
          "NZ_CP069175.1")

y_axis<-rev(c("NZ_CP058229.1",
              "NZ_CP045266.1",
              "NZ_CP028795.1",
              "NZ_CP034127.1",
              "NZ_CP066522.1",
              "NZ_CP069175.1",
              'Patient_115'))

#reshape data: 'wide' --> 'long'
df <- melt(get_lower_tri(matrix_plasmids_ani),na.rm = T)
colnames(df) <- c('var1','var2','ANI')
table(df$var1) 
table(df$var2)


#custom the order so that the plot will display in a lower-left layout
df<-df %>%
  mutate(var1=factor(var1,levels = y_axis),
         var2=factor(var2,levels = x_axis))

#labels to display on the heatmap
txt.df<-data.frame(x=1:8,
                   y=8:1,
                   label=c(
                     "NZ_CP030131.1",
                     "NZ_CP058229.1",
                     "NZ_CP045266.1",
                     "NZ_CP028795.1",
                     "NZ_CP034127.1",
                     "NZ_CP066522.1",
                     "NZ_CP069175.1",
                     'Patient_115'))

df$ANI <- round(df$ANI,3)

p1 <- ggplot(data=df,aes(x=var2,y=var1))+
  geom_tile(aes(fill=ANI),
            color="black")+
  geom_text(aes(label=ANI),size=10) +
  geom_text(data=txt.df,
            aes(x=x,y=y,label=label),size=10,
            nudge_x = 0.45,nudge_y = -0.2)+
  scale_fill_gradient2(low = "#fcbcce",mid = '#f2f2f2',high = "#6cb7d5",midpoint = 99.95) +

  theme_bw()+
  theme(axis.text = element_blank(),
        axis.ticks = element_blank(),
        panel.grid = element_blank(),
        panel.border = element_blank(),
        legend.position = "right",
        legend.title = element_text(size = 24),
        legend.text = element_text(size = 24),
        axis.title = element_blank())+
  coord_cartesian(xlim = c(0,8),y=c(0,8)) +
  guides(fill=guide_colorbar(barheight = 10,
                             ticks.colour = "black"))

```

### Figure 6
### --------------------------------------------------------------------------------------------------------------------------------------------

```{r Fig 6b,c}

df_instrain_comparison = read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Main_figures\instrain_comparisons-SNPs+ANI_groups.csv)")
my_comparisons <- list(c('direct','indirect'))
df_instrain_comparison$ANI <- df_instrain_comparison$ANI *100
df_instrain_comparison$ANI

### Fig 6b
ggviolin(df_instrain_comparison, "Group", "SNPs_number",  fill = "Group",ylim=c(0,35),
         palette = c("#FFE4FA", "#A3C6DE"),
         add = "boxplot", add.params = list(fill = "white")) +
  stat_compare_means(comparisons = my_comparisons,label = "p.label",size=11,
                     vjust=.15,tip.length= c(0,0,0),label.y = 32) +
  scale_x_discrete(labels=c('Direct transmission','Indirect transmission'))+
  labs(x=NULL,y='SNPs number',) +  
  theme_pubr(base_size = 26,legend ='none',x.text.angle = 35) 



###  Fig 6c
###violinplot version (ANI as y-axis)
ggviolin(df_instrain_comparison, "Group", "ANI",  fill = "Group",ylim=c(99.999,100),
         # palette = c("#fde5f0", "#bce3ed"),
         palette = c("#FFE4FA", "#A3C6DE"),
         add = "boxplot", add.params = list(fill = "white")) +
  stat_compare_means(comparisons = my_comparisons,label = "p.label",size=11,
                     vjust=.15,tip.length= c(0,0,0),label.y = 100) +
  scale_x_discrete(labels=c('Direct transmission','Indirect transmission'))+
  labs(x=NULL,y='Average nucleotide identity (%)',) +  
  theme_pubr(base_size = 26,legend ='none',x.text.angle = 30) 

```

### Figure S6
### --------------------------------------------------------------------------------------------------------------------------------------------

```{r Fig s6a}

df_snps_p_corynebacterium  <- read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\corynebacterium-snps_distribution.csv)")
df_snps_p_corynebacterium$position[df_snps_p_corynebacterium$types =='synonymous_variant'] <- df_snps_p_corynebacterium$position[df_snps_p_corynebacterium$types =='synonymous_variant'] - 5000

p_snps_distribution_corynebacterium <- ggplot(df_snps_p_corynebacterium) + geom_segment(aes(x=position,y=0,xend=position,yend=snps_counts,color=types),data=df_snps_p_corynebacterium,size=.3)  +
  scale_color_manual(values=c('nonsynonymous_variant'='red','synonymous_variant'='blue'),
                     name='',
                     labels = c('Nonsynonymous','Synonymous')) + 
  scale_x_continuous(labels= scales::comma)  +
  guides(color=guide_legend(nrow = 2)) + ggtitle('Corynebacterium striatum')

p_snps_distribution_corynebacterium <- ggpar(p_snps_distribution_corynebacterium,
                                             xlab = "Genomic position",
                                             ylab = "SNP number",
                                             # legend.title = "Groups",
                                             font.xtickslab=c(18),
                                             font.ytickslab=c(18),
                                             font.x = c(18,"bold"),
                                             font.y = c(18,"bold"),
                                             font.legend = c(18,'bold')) + theme_pubr(base_size = 20,legend = "none",margin = FALSE,border = T) + 
  ggeasy::easy_center_title() + theme(plot.title = element_text(face = "italic"))

p_snps_distribution_corynebacterium

###-------------------------------------------------------------------------------------------------------------------------------------------------------------------


###snps distribution for Pseudomonas aeruginosa

df_snps_p_pseudomonas  <- read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\pseudomonas-snps_distribution.csv)")
df_snps_p_pseudomonas$position[df_snps_p_pseudomonas$types =='synonymous_variant'] <- df_snps_p_pseudomonas$position[df_snps_p_pseudomonas$types =='synonymous_variant'] - 5000

p_snps_distribution_pseudomonas <- ggplot(df_snps_p_pseudomonas) + geom_segment(aes(x=position,y=0,xend=position,yend=snps_counts,color=types),data=df_snps_p_pseudomonas,size=.3)  +
  scale_color_manual(values=c('nonsynonymous_variant'='red','synonymous_variant'='blue'),
                     name='',
                     labels = c('Nonsynonymous','Synonymous')) + 
  scale_x_continuous(labels= scales::comma)  +
  guides(color=guide_legend(nrow = 2)) + ggtitle('Pseudomonas aeruginosa')

p_snps_distribution_pseudomonas <- ggpar(p_snps_distribution_pseudomonas,
                                         xlab = "Genomic position",
                                         ylab = "SNP number",
                                         # legend.title = "Groups",
                                         font.xtickslab=c(18),
                                         font.ytickslab=c(18),
                                         font.x = c(18,"bold"),
                                         font.y = c(18,"bold"),
                                         font.legend = c(18,'bold')) + theme_pubr(base_size = 20,legend = "none",margin = FALSE,border = T) + 
  ggeasy::easy_center_title() + theme(plot.title = element_text(face = "italic"))

p_snps_distribution_pseudomonas



###-------------------------------------------------------------------------------------------------------------------------------------------------------------------


###snps distribution for stenotrophomonas aeruginosa
df_snps_p_stenotrophomonas  <- read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\stenotrophomonas-snps_distribution.csv)")
df_snps_p_stenotrophomonas$position[df_snps_p_stenotrophomonas$types =='synonymous_variant'] <- df_snps_p_stenotrophomonas$position[df_snps_p_stenotrophomonas$types =='synonymous_variant'] - 5000


p_snps_distribution_stenotrophomonas <- ggplot(df_snps_p_stenotrophomonas) + geom_segment(aes(x=position,y=0,xend=position,yend=snps_counts,color=types),data=df_snps_p_stenotrophomonas,size=.3)  +
  scale_color_manual(values=c('nonsynonymous_variant'='red','synonymous_variant'='blue'),
                     name='',
                     labels = c('Nonsynonymous','Synonymous')) + 
  scale_x_continuous(labels= scales::comma)  +
  guides(color=guide_legend(nrow = 2)) + ggtitle('Stenotrophomonas maltophilia')


p_snps_distribution_stenotrophomonas <- ggpar(p_snps_distribution_stenotrophomonas,
                                              xlab = "Genomic position",
                                              ylab = "SNP number",
                                              # legend.title = "Groups",
                                              font.xtickslab=c(18),
                                              font.ytickslab=c(18),
                                              font.x = c(18,"bold"),
                                              font.y = c(18,"bold"),
                                              font.legend = c(18,'bold')) + theme_pubr(base_size = 20,legend = "none",margin = FALSE,border = T) + 
  ggeasy::easy_center_title() + theme(plot.title = element_text(face = "italic"))

p_snps_distribution_stenotrophomonas


p_snps_distribution_corynebacterium|p_snps_distribution_pseudomonas|p_snps_distribution_stenotrophomonas

```

```{r Fig s6a}

tree_acinetobacter <- read.tree(r'(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\draft+ref_resolved.tre)')
rooted_tree_acinetobacter <- ape::root(tree_acinetobacter, node_number=222, resolve.root = TRUE,outgroup=222)

### display node number for each node
# ggtree(rooted_tree_acinetobacter,layout = 'circular') + geom_text(aes(subset = !isTip, label = node, x = branch), hjust = 1.2, color = "blue")


# Get the external node numbers
external_node_numbers <- 1:length(rooted_tree_acinetobacter$tip.label)
# Get the external node labels
external_node_labels <- rooted_tree_acinetobacter$tip.label
# Combine the node numbers and labels
external_node_df <- data.frame(node = external_node_numbers, label = external_node_labels)
# subset with our draft geomes node number
external_node_df_filtered <- external_node_df %>%
  filter(str_detect(label, 'coassembly'))

###color our draft genomes with their node numbers
nodes_to_color <- c(external_node_df_filtered$node)
# Group the nodes using groupOTU
tree_grouped <- groupOTU(rooted_tree_acinetobacter, nodes_to_color, "group1")

###add colored points for our draft genomes 
# Create ggtree object
ggtree_obj <- ggtree(rooted_tree_acinetobacter)
label_of_interest <- c(external_node_df_filtered$label)
subset_data <- subset(ggtree_obj$data, label %in% label_of_interest)


### circular layout
tree_circular <- ggtree(tree_grouped,layout = 'circular',aes(color=group1)) + #geom_tiplab(size=0,align = T,linetype = 'dashed',linesize = 0.1) +
  geom_tree() + geom_highlight(node=558,alpha=0.3) + #+ geom_tippoint(size=0.1)
  # geom_text2(aes(subset = !is.na(label) & isTip == FALSE, label = label), hjust = -0.5) + #show the bootstrap values
  geom_point(data = subset_data, aes(x, y), color = 'red', size = 1) +
  scale_color_manual(values = c('black','red')) + theme(legend.position = 'none')

```

```{r Fig s6b}

tree_klebsiella <- read.tree(r'(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\draft+ref_resolved.tre)')
rooted_tree_klebsiella <- ape::root(tree_klebsiella, node_number=1316, resolve.root = TRUE,outgroup=1316)

ggtree(rooted_tree_klebsiella,layout = 'circular') + 
  geom_text(aes( label = node, x = branch), hjust = 1.2, color = "blue",size=1) + #show node number
  geom_text2(aes(subset = !is.na(label) & isTip == FALSE, label = label), hjust = -0.5,size=1.5) #show bootstrap values

# Get the external node numbers
external_node_numbers <- 1:length(tree_klebsiella$tip.label)
# Get the external node labels
external_node_labels <- tree_klebsiella$tip.label
# Combine the node numbers and labels
external_node_df <- data.frame(node = external_node_numbers, label = external_node_labels)
# subset with our draft geomes node number
external_node_df_filtered <- external_node_df %>%
  filter(str_detect(label, 'coassembly'))


ggtree(rooted_tree_klebsiella,layout='circular')

###color our draft genomes with their node numbers
nodes_to_color <- c(external_node_df_filtered$node)
# Group the nodes using groupOTU
tree_grouped <- groupOTU(rooted_tree_klebsiella, nodes_to_color, "group1")

###add colored points for our draft genomes 
# Create ggtree object
ggtree_obj <- ggtree(rooted_tree_klebsiella)
label_of_interest <- c(external_node_df_filtered$label)
subset_data <- subset(ggtree_obj$data, label %in% label_of_interest)


### circular layout
tree_circular <- ggtree(tree_grouped,layout = 'circular',aes(color=group1)) + #geom_tiplab(size=0,align = T,linetype = 'dashed',linesize = 0.1) +
  geom_tree() + geom_highlight(node=2102,alpha=0.3) + #+ geom_tippoint(size=0.1)
  # geom_text2(aes(subset = !is.na(label) & isTip == FALSE, label = label), hjust = -0.5) + #show the bootstrap values
  geom_point(data = subset_data, aes(x, y), color = 'red', size = 1) +
  scale_color_manual(values = c('black','red')) + theme(legend.position = 'none') 

```

```{r Fig s6d}

### plot with Phandango

```

```{r Fig s6e,f}

df_recombinations_all <- read.csv(r'(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\klebsiella_recombination-consensus_blocks_all.csv)')
df_pi <- read.table(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\all_patients-klebsiella-windowed.pi)",header = T)


p_recombinations <- ggplot() + 
  geom_segment(aes(x=coordinates,y=y_start,xend=coordinates,yend=y_end,size=cds_length),
               data=df_recombinations_all,color='#377eb8') +
  labs(x='Position on Chromosome',y='Recombination\nblocks') + 
  scale_x_continuous(expand = c(0.02, 0),limits=c(0,5500000),breaks=c(0,2000000,4000000),labels= scales::comma) +
  scale_y_continuous(expand = c(0, 0)) +
  # scale_y_continuous(limits = c(0,23),breaks =seq(1,23),
  #                    labels = c('P123','P143','P136','P146','P77','P25','P52', 'P112','P23',
  #                               'P26','P157','P130','P113','P118','P144','P105','P156','P141',
  #                               'P132','P114','P147','P124','P151')) + 
  # theme_pubr(base_size =30,legend = 'none',border = T) 
  theme_bw(base_size = 35) +
  theme(axis.text.y = element_blank(),
        axis.ticks.y = element_blank(),legend.position = 'none')


p_pi<-ggplot(df_pi,aes(x = BIN_END, y = PI))+
  geom_line(size = 0.66,color='#e41a1c') +
  theme_classic2()+
  theme(panel.spacing = unit(1.5, "cm"),
        strip.background = element_blank(),
        strip.placement = "outside",
        legend.position = "none",
        plot.margin=unit(c(1,1,1,1), "cm"))+
  scale_x_continuous(expand = c(0.02, 0),labels= scales::comma) +
  scale_y_continuous(expand = c(0, 0)) + theme_bw(base_size = 35)
p_pi <- ggpar(p_pi,
              xlab = "",
              ylab= "Pi") 

```

```{r Fig s6g}

df_kegg_pathway = read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\acinetobacter-kegg_pathway-combined.csv)")

cols <- c('Cellular Processes'='#e44235','Environmental Information Processing'='#d92aab',
          'Genetic Information Processing'='#ffc1cc','Metabolism'='#0f52ba',
          'Human Diseases'='#9baddc','Organismal Systems'='#505888')

# calculate the statistics about each category
df_kegg_pathway <- df_kegg_pathway %>% 
  count(level1,level2)
# convert gene numbers to fractions
df_kegg_pathway <- df_kegg_pathway %>%
  mutate(percentage = n/sum(n))


df_kegg_pathway <-df_kegg_pathway %>% 
  mutate(level1 = trimws(str_remove(level1, "(\\s+[A-Za-z]+)?[0-9-]+"))) %>%  
  mutate(level2 = trimws(str_remove(level2, "(\\s+[A-Za-z]+)?[0-9-]+")))


plot_kegg <-ggbarplot(df_kegg_pathway,x = "level2", y = "percentage",orientation = "horiz",
                      fill = "level1",sort.val = 'asc',sort.by.groups = F) + 
  scale_fill_manual(values=cols)
#scale_fill_manual(values=c("#e44235", "#d92aab", "#ffc1cc", "#9baddc", "#0f52ba"))



ggpar(plot_kegg,
      xlab = "",
      ylab = "Proportion",
      legend.title  = 'KEGG Class',
      legend = 'right') + theme_pubr(base_size = 24,border = F,legend = 'none')

```

```{r Fig s6h}


df_kegg_pathway = read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\klebsiella-kegg_pathway-combined.csv)")

cols <- c('Cellular Processes'='#e44235','Environmental Information Processing'='#d92aab',
          'Genetic Information Processing'='#ffc1cc','Metabolism'='#0f52ba',
          'Human Diseases'='#9baddc','Organismal Systems'='#505888')

# calculate the statistics about each category
df_kegg_pathway <- df_kegg_pathway %>% 
  count(level1,level2)
# convert gene numbers to fractions
df_kegg_pathway <- df_kegg_pathway %>%
  mutate(percentage = n/sum(n))


df_kegg_pathway <-df_kegg_pathway %>% 
  mutate(level1 = trimws(str_remove(level1, "(\\s+[A-Za-z]+)?[0-9-]+"))) %>%  
  mutate(level2 = trimws(str_remove(level2, "(\\s+[A-Za-z]+)?[0-9-]+")))


plot_kegg <-ggbarplot(df_kegg_pathway,x = "level2", y = "percentage",orientation = "horiz",
                      fill = "level1",sort.val = 'asc',sort.by.groups = F) + 
  scale_fill_manual(values=cols)
#scale_fill_manual(values=c("#e44235", "#d92aab", "#ffc1cc", "#9baddc", "#0f52ba"))



ggpar(plot_kegg,
      xlab = "",
      ylab = "Proportion",
      legend.title  = 'KEGG Class',
      legend = 'right') + theme_pubr(base_size = 24,border = F,legend = 'none')

```

```{r Fig s7}

df_abundance <- read.csv(r"(LRT_microbiomes_critical_illness\Data_repository\Supplementary_figures\${pid}-abundance_transmitted_strains.csv)") ### replace the pid with P77,P113,P114,P130,P132 to get plot for each individual

df_abundance$Sampling_Day_ICU <- factor(df_abundance$Sampling_Day_ICU,levels = unique(df_abundance$Sampling_Day_ICU))

plot_df <- ggplot(df_abundance, aes(x=Sampling_Day_ICU, y=relative_abundance, group=taxonomy, color=taxonomy)) +
    geom_line(size=1) + 
    geom_point(size=3) +
    ggtitle(label = 'P114' ) + 
    ylim(0,100) +
    theme_bw(base_size = 14) 


ggpar(plot_df,
    ylab = "Relative\nabundance (%)",
    xlab = F,
    x.text.angle = 50)  +
    theme(legend.position = "none") +
    ggeasy::easy_center_title() 

```
